diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/RTMItem.jasm b/src/main/rtm/jp/ngt/rtm/RTMItem.jasm
index 6a22154..d1aa343 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/RTMItem.jasm
+++ b/src/main/rtm/jp/ngt/rtm/RTMItem.jasm
@@ -545,21 +545,21 @@ L_0012:
 L_0018:
     .line 153
     getstatic jp/ngt/rtm/RTMItem/itemtrain Lnet/minecraft/item/Item;
     iconst_3
     ldc "item_train_3"
     invokestatic jp/ngt/rtm/RTMItem/registerItemModel (Lnet/minecraft/item/Item;ILjava/lang/String;)V
 L_001e:
     .line 154
     getstatic jp/ngt/rtm/RTMItem/itemtrain Lnet/minecraft/item/Item;
     bipush 127
-    ldc "item_train_127"
+    ldc "item_train_fixrtm_test"
     invokestatic jp/ngt/rtm/RTMItem/registerItemModel (Lnet/minecraft/item/Item;ILjava/lang/String;)V
 L_0024:
     .line 155
     getstatic jp/ngt/rtm/RTMItem/itemMotorman Lnet/minecraft/item/Item;
     iconst_0
     ldc "item_npc_0"
     invokestatic jp/ngt/rtm/RTMItem/registerItemModel (Lnet/minecraft/item/Item;ILjava/lang/String;)V
 L_002a:
     .line 156
     getstatic jp/ngt/rtm/RTMItem/itemMotorman Lnet/minecraft/item/Item;
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/train/EntityBogie.jasm b/src/main/rtm/jp/ngt/rtm/entity/train/EntityBogie.jasm
index e0457b7..6a0fe61 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/train/EntityBogie.jasm
+++ b/src/main/rtm/jp/ngt/rtm/entity/train/EntityBogie.jasm
@@ -1470,24 +1470,34 @@ L_0012:
     .end stack
     iconst_0
     istore 1
 L_0017:
     .line 376
     .stack use locals
     .end stack
     aload 0
     invokevirtual jp/ngt/rtm/entity/train/EntityBogie/getTrain ()Ljp/ngt/rtm/entity/train/EntityTrainBase;
     ifnonnull L_0021
+
+    .line 1378
+    aload 0
+    getfield jp/ngt/rtm/entity/train/EntityBogie/world Lnet/minecraft/world/World;
+    getfield net/minecraft/world/World/isRemote Z
+    ifne L_ifRemote
+
 L_001d:
     .line 378
     aload 0
     invokevirtual jp/ngt/rtm/entity/train/EntityBogie/setDead ()V
+
+L_ifRemote:
+
 L_0021:
     .line 380
     .stack use locals
     .end stack
     return
 L_0025:
 .end method
 
 .method public onBogieUpdate ()V
     .limit stack 2
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.jasm b/src/main/rtm/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.jasm
index 4d94e4a..18e2380 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.jasm
+++ b/src/main/rtm/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.jasm
@@ -931,48 +931,50 @@ L_0004:
     astore 4
 L_000b:
     .line 364
     .var 4 is set Ljp/ngt/rtm/modelpack/modelset/ModelSetBase; from L_000b to L_0011
     aload 4
     invokevirtual jp/ngt/rtm/modelpack/modelset/ModelSetBase/getCollisionObj ()Ljp/ngt/rtm/entity/util/CollisionObj;
     invokevirtual jp/ngt/rtm/entity/util/CollisionObj/getSizeBox ()Lnet/minecraft/util/math/AxisAlignedBB;
     astore 3
 L_0011:
     .line 370
-    goto L_0028
+    goto L_end_try
 L_0014:
     .line 366
     .stack
         locals Object jp/ngt/rtm/entity/vehicle/EntityVehicleBase
         locals Object net/minecraft/entity/Entity
         locals Object net/minecraft/entity/Entity
         locals Object net/minecraft/util/math/AxisAlignedBB
         stacks Object java/lang/NullPointerException
     .end stack
-    astore 4
-L_0018:
-    .line 368
-    .var 4 is e Ljava/lang/NullPointerException; from L_0018 to L_0028
-    aload 4
-    invokevirtual java/lang/NullPointerException/printStackTrace ()V
+    pop
+L_end_try:
+    .stack use locals
+    .end stack
+    aload 3
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/fixRiderPosOnDismount_remakeAABB (Lnet/minecraft/util/math/AxisAlignedBB;)Z
+    ifeq L_aabb_remake_end
 L_001c:
     .line 369
     new net/minecraft/util/math/AxisAlignedBB
     dup
     ldc -1.5
     dconst_0
     ldc -2.0
     ldc 1.5
     ldc 3.0
     ldc 2.0
     invokespecial net/minecraft/util/math/AxisAlignedBB/<init> (DDDDDD)V
     astore 3
+L_aabb_remake_end:
 L_0028:
     .line 374
     .stack use locals
     .end stack
     aload 2
     getfield net/minecraft/entity/Entity/posX D
     aload 0
     getfield jp/ngt/rtm/entity/vehicle/EntityVehicleBase/posX D
     dsub
     aload 2
@@ -1091,21 +1093,21 @@ L_0092:
     .line 386
     .var 12 is z D from L_0092 to L_00b1
     aload 0
     getfield jp/ngt/rtm/entity/vehicle/EntityVehicleBase/world Lnet/minecraft/world/World;
     new net/minecraft/util/math/BlockPos
     dup
     dload 8
     dload 10
     dload 12
     invokespecial net/minecraft/util/math/BlockPos/<init> (DDD)V
-    invokevirtual net/minecraft/world/World/getTopSolidOrLiquidBlock (Lnet/minecraft/util/math/BlockPos;)Lnet/minecraft/util/math/BlockPos;
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/fixRiderPosOnDismount_calcBestBlockPos (Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)Lnet/minecraft/util/math/BlockPos;
     astore 14
 L_009e:
     .line 387
     .var 14 is pos Lnet/minecraft/util/math/BlockPos; from L_009e to L_00b1
     aload 1
     aload 14
     invokevirtual net/minecraft/util/math/BlockPos/getX ()I
     i2d
     ldc 0.5
     dadd
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/vehicle/WeatherEffectDummy.jasm b/src/main/rtm/jp/ngt/rtm/entity/vehicle/WeatherEffectDummy.jasm
index 5ffef9f..3e50be4 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/vehicle/WeatherEffectDummy.jasm
+++ b/src/main/rtm/jp/ngt/rtm/entity/vehicle/WeatherEffectDummy.jasm
@@ -223,13 +223,25 @@ L_0003:
     .limit stack 0
     .limit local 2
 L_0000:
     .line 85
     .var 0 is this Ljp/ngt/rtm/entity/vehicle/WeatherEffectDummy; from L_0000 to L_0003
     .var 1 is compound Lnet/minecraft/nbt/NBTTagCompound; from L_0000 to L_0003
     return
 L_0003:
 .end method
 
+.method public getEntityBoundingBox ()Lnet/minecraft/util/math/AxisAlignedBB;
+    .limit stack 1
+    .limit local 1
+L_start:
+    .var 0 is this Ljp/ngt/rtm/entity/vehicle/WeatherEffectDummy; from L_start to L_end
+    aload 0
+    getfield jp/ngt/rtm/entity/vehicle/WeatherEffectDummy/parent Ljp/ngt/rtm/entity/vehicle/EntityVehicleBase;
+    invokevirtual net/minecraft/entity/Entity/getEntityBoundingBox ()Lnet/minecraft/util/math/AxisAlignedBB;
+    areturn
+L_end:
+.end method
+
 .field private final parent Ljp/ngt/rtm/entity/vehicle/EntityVehicleBase;
 .end field
 
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/ModelPackManager.jasm b/src/main/rtm/jp/ngt/rtm/modelpack/ModelPackManager.jasm
index 0866a89..1ed0e7e 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/ModelPackManager.jasm
+++ b/src/main/rtm/jp/ngt/rtm/modelpack/ModelPackManager.jasm
@@ -902,26 +902,34 @@ L_0048:
 L_004b:
     .line 260
     .stack
         locals Object jp/ngt/rtm/modelpack/ModelPackManager
         locals Object net/minecraft/entity/player/EntityPlayerMP
         locals Integer
         locals Object java/util/Iterator
     .end stack
     goto L_000b
 L_004f:
-    .line 261
+    .line 1261
     .stack
         locals Object jp/ngt/rtm/modelpack/ModelPackManager
         locals Object net/minecraft/entity/player/EntityPlayerMP
         locals Integer
     .end stack
+
+    new com/anatawa12/fixRtm/network/SentAllModels
+    dup
+    invokespecial com/anatawa12/fixRtm/network/SentAllModels/<init> ()V
+    aload 1
+    invokestatic com/anatawa12/fixRtm/network/NetworkHandler/sendPacketEPM (Lnet/minecraftforge/fml/common/network/simpleimpl/IMessage;Lnet/minecraft/entity/player/EntityPlayerMP;)V
+
+    .line 261
     return
 L_0053:
 .end method
 
 .method public addModelSetName (ILjava/lang/String;Ljava/lang/String;)V
     .limit stack 3
     .limit local 6
 L_0000:
     .line 266
     .var 0 is this Ljp/ngt/rtm/modelpack/ModelPackManager; from L_0000 to L_0068
@@ -1136,20 +1144,21 @@ L_002e:
 .method public getResourceSet (Ljp/ngt/rtm/modelpack/ResourceType;Ljava/lang/String;)Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
     .signature "<T:Ljp/ngt/rtm/modelpack/modelset/ResourceSet;>(Ljp/ngt/rtm/modelpack/ResourceType;Ljava/lang/String;)TT;"
     .limit stack 4
     .limit local 7
 L_0000:
     .line 310
     .var 0 is this Ljp/ngt/rtm/modelpack/ModelPackManager; from L_0000 to L_005d
     .var 1 is type Ljp/ngt/rtm/modelpack/ResourceType; from L_0000 to L_005d
     .var 2 is name Ljava/lang/String; from L_0000 to L_005d
     aload 1
+    invokestatic jp/ngt/rtm/modelpack/ModelPackManager/"get parent or me" (Ljp/ngt/rtm/modelpack/ResourceType;)Ljp/ngt/rtm/modelpack/ResourceType;
     astore 3
 L_0004:
     .line 311
     .var 3 is parent Ljp/ngt/rtm/modelpack/ResourceType; from L_0004 to L_005d
     aload 1
     getfield jp/ngt/rtm/modelpack/ResourceType/parent Ljp/ngt/rtm/modelpack/ResourceType;
     ifnull L_000e
 L_0009:
     .line 313
     aload 1
@@ -1212,64 +1221,34 @@ L_002b:
     invokeinterface java/util/Map/get (Ljava/lang/Object;)Ljava/lang/Object;
     checkcast jp/ngt/rtm/modelpack/modelset/ResourceSet
     astore 6
 L_0035:
     .line 319
     .var 6 is modelSet Ljp/ngt/rtm/modelpack/modelset/ResourceSet; signature "TT;" from L_0035 to L_005d
     aload 6
     ifnonnull L_0058
 L_0039:
     .line 321
-    aload 0
-    getfield jp/ngt/rtm/modelpack/ModelPackManager/dummyMap Ljava/util/Map;
-    aload 1
-    getfield jp/ngt/rtm/modelpack/ResourceType/hasSubType Z
-    ifeq L_0043
     aload 1
-    getfield jp/ngt/rtm/modelpack/ResourceType/subType Ljava/lang/String;
-    goto L_0047
-L_0043:
+    aload 2
+    invokestatic com/anatawa12/fixRtm/DummyModelPackManager/getSet (Ljp/ngt/rtm/modelpack/ResourceType;Ljava/lang/String;)Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
+    areturn
+L_0058:
+    .line 327
     .stack
         locals Object jp/ngt/rtm/modelpack/ModelPackManager
         locals Object jp/ngt/rtm/modelpack/ResourceType
         locals Object java/lang/String
         locals Object jp/ngt/rtm/modelpack/ResourceType
         locals Integer
         locals Object java/util/Map
         locals Object jp/ngt/rtm/modelpack/modelset/ResourceSet
-        stacks Object java/util/Map
-    .end stack
-    aload 1
-    getfield jp/ngt/rtm/modelpack/ResourceType/name Ljava/lang/String;
-L_0047:
-    .stack use locals
-        stacks Object java/util/Map
-        stacks Object java/lang/Object
-    .end stack
-    invokeinterface java/util/Map/get (Ljava/lang/Object;)Ljava/lang/Object;
-    checkcast jp/ngt/rtm/modelpack/modelset/ResourceSet
-    astore 6
-L_004c:
-    .line 322
-    aload 6
-    ifnonnull L_0058
-L_0050:
-    .line 324
-    new jp/ngt/rtm/modelpack/ModelPackException
-    dup
-    ldc "Default model is not registered."
-    aload 2
-    invokespecial jp/ngt/rtm/modelpack/ModelPackException/<init> (Ljava/lang/String;Ljava/lang/String;)V
-    athrow
-L_0058:
-    .line 327
-    .stack use locals
     .end stack
     aload 6
     areturn
 L_005d:
 .end method
 
 .method public getModelList (Ljp/ngt/rtm/modelpack/ResourceType;)Ljava/util/List;
     .signature "(Ljp/ngt/rtm/modelpack/ResourceType;)Ljava/util/List<Ljp/ngt/rtm/modelpack/modelset/ResourceSet;>;"
     .limit stack 3
     .limit local 4
@@ -2052,29 +2031,29 @@ L_0018:
 .field private static final SC_INCLUDE Ljava/util/regex/Pattern;
 .end field
 
 .field private static final ENCODING "[Ljava/lang/String;"
 .end field
 
 .field private final typeMap Ljava/util/Map;
     .signature "Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/modelpack/ResourceType;>;"
 .end field
 
-.field private final allModelSetMap Ljava/util/Map;
+.field public final allModelSetMap Ljava/util/Map;
     .signature "Ljava/util/Map<Ljp/ngt/rtm/modelpack/ResourceType;Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/modelpack/modelset/ResourceSet;>;>;"
 .end field
 
 .field private final smpModelSetMap Ljava/util/Map;
     .signature "Ljava/util/Map<Ljp/ngt/rtm/modelpack/ResourceType;Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/modelpack/modelset/ResourceSet;>;>;"
 .end field
 
-.field private final dummyMap Ljava/util/Map;
+.field public final dummyMap Ljava/util/Map;
     .signature "Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/modelpack/modelset/ResourceSet;>;"
 .end field
 
 .field public final unconstructSets Ljava/util/List;
     .signature "Ljava/util/List<Ljp/ngt/rtm/modelpack/modelset/ResourceSet;>;"
 .end field
 
 .field private final modelCache Ljava/util/Map;
     .signature "Ljava/util/Map<Ljava/lang/String;Ljp/ngt/ngtlib/renderer/model/IModelNGT;>;"
 .end field
@@ -2093,10 +2072,31 @@ L_0018:
 
 .field public modelLoaded Z
 .end field
 
 .field private static final DEFAULT_DOMAIN Ljava/lang/String; = "minecraft"
 .end field
 
 .field static final synthetic $assertionsDisabled Z
 .end field
 
+.method public static "get parent or me" "(Ljp/ngt/rtm/modelpack/ResourceType;)Ljp/ngt/rtm/modelpack/ResourceType;"
+    .limit local 1
+    .limit stack 1
+
+    .stack
+        locals Object jp/ngt/rtm/modelpack/ResourceType
+    .end stack
+    aload 0
+    getfield jp/ngt/rtm/modelpack/ResourceType/parent Ljp/ngt/rtm/modelpack/ResourceType;
+    ifnull L_if_null_to
+    .stack use locals
+    .end stack
+    aload 0
+    getfield jp/ngt/rtm/modelpack/ResourceType/parent Ljp/ngt/rtm/modelpack/ResourceType;
+    areturn
+L_if_null_to:
+    .stack use locals
+    .end stack
+    aload 0
+    areturn
+.end method
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/init/ModelPackConstructThread.jasm b/src/main/rtm/jp/ngt/rtm/modelpack/init/ModelPackConstructThread.jasm
index a9e7430..646cbfb 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/init/ModelPackConstructThread.jasm
+++ b/src/main/rtm/jp/ngt/rtm/modelpack/init/ModelPackConstructThread.jasm
@@ -1,13 +1,13 @@
 .bytecode 52.0
 .source "ModelPackConstructThread.java"
-.class public final jp/ngt/rtm/modelpack/init/ModelPackConstructThread
+.class public jp/ngt/rtm/modelpack/init/ModelPackConstructThread
 .super java/lang/Thread
 .inner public static final enum jp/ngt/rtm/modelpack/init/ProgressStateHolder$ProgressState inner ProgressState outer jp/ngt/rtm/modelpack/init/ProgressStateHolder
 
 .method public <init> (Lnet/minecraftforge/fml/relauncher/Side;Ljp/ngt/rtm/modelpack/init/ModelPackLoadThread;)V
     .limit stack 2
     .limit local 3
 L_0000:
     .line 22
     .var 0 is this Ljp/ngt/rtm/modelpack/init/ModelPackConstructThread; from L_0000 to L_0017
     .var 1 is par1 Lnet/minecraftforge/fml/relauncher/Side; from L_0000 to L_0017
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/init/ModelPackLoadThread.jasm b/src/main/rtm/jp/ngt/rtm/modelpack/init/ModelPackLoadThread.jasm
index 85ac6d0..a8c04bf 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/init/ModelPackLoadThread.jasm
+++ b/src/main/rtm/jp/ngt/rtm/modelpack/init/ModelPackLoadThread.jasm
@@ -517,45 +517,45 @@ L_0035:
     iconst_0
     invokestatic jp/ngt/rtm/modelpack/init/ProgressStateHolder$ProgressState/values "()[Ljp/ngt/rtm/modelpack/init/ProgressStateHolder\$ProgressState;"
     arraylength
     ldc ""
     invokevirtual jp/ngt/rtm/modelpack/init/ModelPackLoadThread/setBarMaxValue (IILjava/lang/String;)V
 L_003e:
     .line 184
     invokestatic jp/ngt/ngtlib/io/NGTLog/startTimer ()V
 L_0041:
     .line 185
-    new jp/ngt/rtm/modelpack/init/ModelPackConstructThread
+    new com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread
     dup
     aload 0
     getfield jp/ngt/rtm/modelpack/init/ModelPackLoadThread/threadSide Lnet/minecraftforge/fml/relauncher/Side;
     aload 0
-    invokespecial jp/ngt/rtm/modelpack/init/ModelPackConstructThread/<init> (Lnet/minecraftforge/fml/relauncher/Side;Ljp/ngt/rtm/modelpack/init/ModelPackLoadThread;)V
+    invokespecial com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread/<init> (Lnet/minecraftforge/fml/relauncher/Side;Ljp/ngt/rtm/modelpack/init/ModelPackLoadThread;)V
     astore 1
 L_004a:
     .line 186
-    .var 1 is thread2 Ljp/ngt/rtm/modelpack/init/ModelPackConstructThread; from L_004a to L_006a
+    .var 1 is thread2 Lcom/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread; from L_004a to L_006a
     aload 1
-    invokevirtual jp/ngt/rtm/modelpack/init/ModelPackConstructThread/start ()V
+    invokevirtual com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread/start ()V
 L_004e:
     .line 187
     getstatic jp/ngt/rtm/modelpack/ModelPackManager/INSTANCE Ljp/ngt/rtm/modelpack/ModelPackManager;
     aload 0
     invokevirtual jp/ngt/rtm/modelpack/ModelPackManager/load (Ljp/ngt/rtm/modelpack/init/ModelPackLoadThread;)V
 L_0053:
     .line 188
     .stack
         locals Object jp/ngt/rtm/modelpack/init/ModelPackLoadThread
-        locals Object jp/ngt/rtm/modelpack/init/ModelPackConstructThread
+        locals Object com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread
     .end stack
     aload 1
-    invokevirtual jp/ngt/rtm/modelpack/init/ModelPackConstructThread/setFinish ()Z
+    invokevirtual com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread/setFinish ()Z
     ifne L_005e
 L_0059:
     .line 190
     ldc 500L
     invokestatic jp/ngt/rtm/modelpack/init/ModelPackLoadThread/sleep (J)V
     goto L_0053
 L_005e:
     .line 192
     .stack use locals
     .end stack
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/state/ResourceState.jasm b/src/main/rtm/jp/ngt/rtm/modelpack/state/ResourceState.jasm
index ce341fe..c56013a 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/state/ResourceState.jasm
+++ b/src/main/rtm/jp/ngt/rtm/modelpack/state/ResourceState.jasm
@@ -324,20 +324,23 @@ L_0009:
     .stack
         locals Object jp/ngt/rtm/modelpack/state/ResourceState
     .end stack
     aload 0
     getstatic jp/ngt/rtm/modelpack/ModelPackManager/INSTANCE Ljp/ngt/rtm/modelpack/ModelPackManager;
     aload 0
     getfield jp/ngt/rtm/modelpack/state/ResourceState/type Ljp/ngt/rtm/modelpack/ResourceType;
     aload 0
     getfield jp/ngt/rtm/modelpack/state/ResourceState/modelName Ljava/lang/String;
     invokevirtual jp/ngt/rtm/modelpack/ModelPackManager/getResourceSet (Ljp/ngt/rtm/modelpack/ResourceType;Ljava/lang/String;)Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
+    aload 0
+    getfield jp/ngt/rtm/modelpack/state/ResourceState/type Ljp/ngt/rtm/modelpack/ResourceType;
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/eraseNullForModelSet (Ljp/ngt/rtm/modelpack/modelset/ResourceSet;Ljp/ngt/rtm/modelpack/ResourceType;)Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
     putfield jp/ngt/rtm/modelpack/state/ResourceState/modelSet Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
 L_0014:
     .line 118
     aload 0
     getfield jp/ngt/rtm/modelpack/state/ResourceState/modelSet Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
     invokevirtual jp/ngt/rtm/modelpack/modelset/ResourceSet/isDummy ()Z
     ifne L_0022
 L_001a:
     .line 120
     aload 0
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/rail/BlockMarker.jasm b/src/main/rtm/jp/ngt/rtm/rail/BlockMarker.jasm
index 8663826..e509f2d 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/rail/BlockMarker.jasm
+++ b/src/main/rtm/jp/ngt/rtm/rail/BlockMarker.jasm
@@ -565,20 +565,22 @@ L_0026:
     .limit local 12
 L_0000:
     .line 190
     .var 0 is this Ljp/ngt/rtm/rail/BlockMarker; from L_0000 to L_0046
     .var 1 is world Lnet/minecraft/world/World; from L_0000 to L_0046
     .var 2 is x I from L_0000 to L_0046
     .var 3 is y I from L_0000 to L_0046
     .var 4 is z I from L_0000 to L_0046
     .var 5 is player Lnet/minecraft/entity/player/EntityPlayer; from L_0000 to L_0046
     .var 6 is makeRail Z from L_0000 to L_0046
+    aload 5
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/BlockMarker_onMarkerActivated "(Lnet/minecraft/entity/player/EntityPlayer;)V"
     aload 0
     aload 5
     iload 6
     invokevirtual jp/ngt/rtm/rail/BlockMarker/hasRail (Lnet/minecraft/entity/player/EntityPlayer;Z)Ljp/ngt/rtm/modelpack/state/ResourceStateRail;
     astore 7
 L_0007:
     .line 191
     .var 7 is prop Ljp/ngt/rtm/modelpack/state/ResourceStateRail; from L_0007 to L_0046
     aload 7
     ifnull L_0041
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/rail/util/RailMaker.jasm b/src/main/rtm/jp/ngt/rtm/rail/util/RailMaker.jasm
index 61097af..41d3cbc 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/rail/util/RailMaker.jasm
+++ b/src/main/rtm/jp/ngt/rtm/rail/util/RailMaker.jasm
@@ -502,21 +502,21 @@ L_005a:
     aload 2
     getfield jp/ngt/rtm/rail/util/RailPosition/blockY I
     invokestatic java/lang/Integer/valueOf (I)Ljava/lang/Integer;
     aastore
     dup
     iconst_2
     aload 2
     getfield jp/ngt/rtm/rail/util/RailPosition/blockZ I
     invokestatic java/lang/Integer/valueOf (I)Ljava/lang/Integer;
     aastore
-    invokestatic jp/ngt/ngtlib/io/NGTLog/sendChatMessageToAll "(Ljava/lang/String;[Ljava/lang/Object;)V"
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/sendSwitchTypeError "(Ljava/lang/String;[Ljava/lang/Object;)V"
 L_0072:
     .line 101
     .stack use locals
     .end stack
     aconst_null
     areturn
 L_0077:
 .end method
 
 .field private worldObj Lnet/minecraft/world/World;
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/sound/MovingSoundMaker.jasm b/src/main/rtm/jp/ngt/rtm/sound/MovingSoundMaker.jasm
index 55fb34a..2b55a66 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/sound/MovingSoundMaker.jasm
+++ b/src/main/rtm/jp/ngt/rtm/sound/MovingSoundMaker.jasm
@@ -310,20 +310,22 @@ L_0010:
     invokestatic jp/ngt/ngtlib/io/NGTJson/getGson ()Lcom/google/gson/Gson;
     aload 2
     getstatic jp/ngt/rtm/sound/MovingSoundMaker/TYPE Ljava/lang/reflect/ParameterizedType;
     invokevirtual com/google/gson/Gson/fromJson (Ljava/lang/String;Ljava/lang/reflect/Type;)Ljava/lang/Object;
     checkcast java/util/Map
     astore 3
 L_0018:
     .line 115
     .var 3 is map Ljava/util/Map; signature "Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/sound/MovingSoundMaker\$SoundListDummy;>;" from L_0018 to L_009c
     aload 3
+    aload 0
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/MovingSoundMaker_loadSoundJson_nullCheck (Ljava/util/Map;Ljava/lang/String;)Ljava/util/Map;
     invokeinterface java/util/Map/entrySet ()Ljava/util/Set;
     invokeinterface java/util/Set/iterator ()Ljava/util/Iterator;
     astore 4
 L_001e:
     .stack
         locals Object java/lang/String
         locals Object java/util/Map
         locals Object java/lang/String
         locals Object java/util/Map
         locals Object java/util/Iterator
