diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/RTMItem.jasm b/src/main/rtm/jp/ngt/rtm/RTMItem.jasm
index 6a22154..d1aa343 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/RTMItem.jasm
+++ b/src/main/rtm/jp/ngt/rtm/RTMItem.jasm
@@ -552,7 +552,7 @@ L_001e:
     .line 154
     getstatic jp/ngt/rtm/RTMItem/itemtrain Lnet/minecraft/item/Item;
     bipush 127
-    ldc "item_train_127"
+    ldc "item_train_fixrtm_test"
     invokestatic jp/ngt/rtm/RTMItem/registerItemModel (Lnet/minecraft/item/Item;ILjava/lang/String;)V
 L_0024:
     .line 155
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/train/EntityBogie.jasm b/src/main/rtm/jp/ngt/rtm/entity/train/EntityBogie.jasm
index e0457b7..6a0fe61 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/train/EntityBogie.jasm
+++ b/src/main/rtm/jp/ngt/rtm/entity/train/EntityBogie.jasm
@@ -1477,10 +1477,20 @@ L_0017:
     aload 0
     invokevirtual jp/ngt/rtm/entity/train/EntityBogie/getTrain ()Ljp/ngt/rtm/entity/train/EntityTrainBase;
     ifnonnull L_0021
+
+    .line 1378
+    aload 0
+    getfield jp/ngt/rtm/entity/train/EntityBogie/world Lnet/minecraft/world/World;
+    getfield net/minecraft/world/World/isRemote Z
+    ifne L_ifRemote
+
 L_001d:
     .line 378
     aload 0
     invokevirtual jp/ngt/rtm/entity/train/EntityBogie/setDead ()V
+
+L_ifRemote:
+
 L_0021:
     .line 380
     .stack use locals
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.jasm b/src/main/rtm/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.jasm
index 4d94e4a..18e2380 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.jasm
+++ b/src/main/rtm/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.jasm
@@ -938,7 +938,7 @@ L_000b:
     astore 3
 L_0011:
     .line 370
-    goto L_0028
+    goto L_end_try
 L_0014:
     .line 366
     .stack
@@ -948,12 +948,13 @@ L_0014:
         locals Object net/minecraft/util/math/AxisAlignedBB
         stacks Object java/lang/NullPointerException
     .end stack
-    astore 4
-L_0018:
-    .line 368
-    .var 4 is e Ljava/lang/NullPointerException; from L_0018 to L_0028
-    aload 4
-    invokevirtual java/lang/NullPointerException/printStackTrace ()V
+    pop
+L_end_try:
+    .stack use locals
+    .end stack
+    aload 3
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/fixRiderPosOnDismount_remakeAABB (Lnet/minecraft/util/math/AxisAlignedBB;)Z
+    ifeq L_aabb_remake_end
 L_001c:
     .line 369
     new net/minecraft/util/math/AxisAlignedBB
@@ -966,6 +967,7 @@ L_001c:
     ldc 2.0
     invokespecial net/minecraft/util/math/AxisAlignedBB/<init> (DDDDDD)V
     astore 3
+L_aabb_remake_end:
 L_0028:
     .line 374
     .stack use locals
@@ -1098,7 +1100,7 @@ L_0092:
     dload 10
     dload 12
     invokespecial net/minecraft/util/math/BlockPos/<init> (DDD)V
-    invokevirtual net/minecraft/world/World/getTopSolidOrLiquidBlock (Lnet/minecraft/util/math/BlockPos;)Lnet/minecraft/util/math/BlockPos;
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/fixRiderPosOnDismount_calcBestBlockPos (Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)Lnet/minecraft/util/math/BlockPos;
     astore 14
 L_009e:
     .line 387
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/vehicle/WeatherEffectDummy.jasm b/src/main/rtm/jp/ngt/rtm/entity/vehicle/WeatherEffectDummy.jasm
index 5ffef9f..3e50be4 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/entity/vehicle/WeatherEffectDummy.jasm
+++ b/src/main/rtm/jp/ngt/rtm/entity/vehicle/WeatherEffectDummy.jasm
@@ -230,6 +230,18 @@ L_0000:
 L_0003:
 .end method
 
+.method public getEntityBoundingBox ()Lnet/minecraft/util/math/AxisAlignedBB;
+    .limit stack 1
+    .limit local 1
+L_start:
+    .var 0 is this Ljp/ngt/rtm/entity/vehicle/WeatherEffectDummy; from L_start to L_end
+    aload 0
+    getfield jp/ngt/rtm/entity/vehicle/WeatherEffectDummy/parent Ljp/ngt/rtm/entity/vehicle/EntityVehicleBase;
+    invokevirtual net/minecraft/entity/Entity/getEntityBoundingBox ()Lnet/minecraft/util/math/AxisAlignedBB;
+    areturn
+L_end:
+.end method
+
 .field private final parent Ljp/ngt/rtm/entity/vehicle/EntityVehicleBase;
 .end field
 
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/ModelPackManager.jasm b/src/main/rtm/jp/ngt/rtm/modelpack/ModelPackManager.jasm
index 0866a89..1ed0e7e 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/ModelPackManager.jasm
+++ b/src/main/rtm/jp/ngt/rtm/modelpack/ModelPackManager.jasm
@@ -909,12 +909,20 @@ L_004b:
     .end stack
     goto L_000b
 L_004f:
-    .line 261
+    .line 1261
     .stack
         locals Object jp/ngt/rtm/modelpack/ModelPackManager
         locals Object net/minecraft/entity/player/EntityPlayerMP
         locals Integer
     .end stack
+
+    new com/anatawa12/fixRtm/network/SentAllModels
+    dup
+    invokespecial com/anatawa12/fixRtm/network/SentAllModels/<init> ()V
+    aload 1
+    invokestatic com/anatawa12/fixRtm/network/NetworkHandler/sendPacketEPM (Lnet/minecraftforge/fml/common/network/simpleimpl/IMessage;Lnet/minecraft/entity/player/EntityPlayerMP;)V
+
+    .line 261
     return
 L_0053:
 .end method
@@ -1143,6 +1151,7 @@ L_0000:
     .var 1 is type Ljp/ngt/rtm/modelpack/ResourceType; from L_0000 to L_005d
     .var 2 is name Ljava/lang/String; from L_0000 to L_005d
     aload 1
+    invokestatic jp/ngt/rtm/modelpack/ModelPackManager/"get parent or me" (Ljp/ngt/rtm/modelpack/ResourceType;)Ljp/ngt/rtm/modelpack/ResourceType;
     astore 3
 L_0004:
     .line 311
@@ -1219,15 +1228,12 @@ L_0035:
     ifnonnull L_0058
 L_0039:
     .line 321
-    aload 0
-    getfield jp/ngt/rtm/modelpack/ModelPackManager/dummyMap Ljava/util/Map;
-    aload 1
-    getfield jp/ngt/rtm/modelpack/ResourceType/hasSubType Z
-    ifeq L_0043
     aload 1
-    getfield jp/ngt/rtm/modelpack/ResourceType/subType Ljava/lang/String;
-    goto L_0047
-L_0043:
+    aload 2
+    invokestatic com/anatawa12/fixRtm/DummyModelPackManager/getSet (Ljp/ngt/rtm/modelpack/ResourceType;Ljava/lang/String;)Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
+    areturn
+L_0058:
+    .line 327
     .stack
         locals Object jp/ngt/rtm/modelpack/ModelPackManager
         locals Object jp/ngt/rtm/modelpack/ResourceType
@@ -1236,33 +1242,6 @@ L_0043:
         locals Integer
         locals Object java/util/Map
         locals Object jp/ngt/rtm/modelpack/modelset/ResourceSet
-        stacks Object java/util/Map
-    .end stack
-    aload 1
-    getfield jp/ngt/rtm/modelpack/ResourceType/name Ljava/lang/String;
-L_0047:
-    .stack use locals
-        stacks Object java/util/Map
-        stacks Object java/lang/Object
-    .end stack
-    invokeinterface java/util/Map/get (Ljava/lang/Object;)Ljava/lang/Object;
-    checkcast jp/ngt/rtm/modelpack/modelset/ResourceSet
-    astore 6
-L_004c:
-    .line 322
-    aload 6
-    ifnonnull L_0058
-L_0050:
-    .line 324
-    new jp/ngt/rtm/modelpack/ModelPackException
-    dup
-    ldc "Default model is not registered."
-    aload 2
-    invokespecial jp/ngt/rtm/modelpack/ModelPackException/<init> (Ljava/lang/String;Ljava/lang/String;)V
-    athrow
-L_0058:
-    .line 327
-    .stack use locals
     .end stack
     aload 6
     areturn
@@ -2059,7 +2038,7 @@ L_0018:
     .signature "Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/modelpack/ResourceType;>;"
 .end field
 
-.field private final allModelSetMap Ljava/util/Map;
+.field public final allModelSetMap Ljava/util/Map;
     .signature "Ljava/util/Map<Ljp/ngt/rtm/modelpack/ResourceType;Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/modelpack/modelset/ResourceSet;>;>;"
 .end field
 
@@ -2067,7 +2046,7 @@ L_0018:
     .signature "Ljava/util/Map<Ljp/ngt/rtm/modelpack/ResourceType;Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/modelpack/modelset/ResourceSet;>;>;"
 .end field
 
-.field private final dummyMap Ljava/util/Map;
+.field public final dummyMap Ljava/util/Map;
     .signature "Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/modelpack/modelset/ResourceSet;>;"
 .end field
 
@@ -2100,3 +2079,24 @@ L_0018:
 .field static final synthetic $assertionsDisabled Z
 .end field
 
+.method public static "get parent or me" "(Ljp/ngt/rtm/modelpack/ResourceType;)Ljp/ngt/rtm/modelpack/ResourceType;"
+    .limit local 1
+    .limit stack 1
+
+    .stack
+        locals Object jp/ngt/rtm/modelpack/ResourceType
+    .end stack
+    aload 0
+    getfield jp/ngt/rtm/modelpack/ResourceType/parent Ljp/ngt/rtm/modelpack/ResourceType;
+    ifnull L_if_null_to
+    .stack use locals
+    .end stack
+    aload 0
+    getfield jp/ngt/rtm/modelpack/ResourceType/parent Ljp/ngt/rtm/modelpack/ResourceType;
+    areturn
+L_if_null_to:
+    .stack use locals
+    .end stack
+    aload 0
+    areturn
+.end method
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/init/ModelPackConstructThread.jasm b/src/main/rtm/jp/ngt/rtm/modelpack/init/ModelPackConstructThread.jasm
index a9e7430..646cbfb 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/init/ModelPackConstructThread.jasm
+++ b/src/main/rtm/jp/ngt/rtm/modelpack/init/ModelPackConstructThread.jasm
@@ -1,6 +1,6 @@
 .bytecode 52.0
 .source "ModelPackConstructThread.java"
-.class public final jp/ngt/rtm/modelpack/init/ModelPackConstructThread
+.class public jp/ngt/rtm/modelpack/init/ModelPackConstructThread
 .super java/lang/Thread
 .inner public static final enum jp/ngt/rtm/modelpack/init/ProgressStateHolder$ProgressState inner ProgressState outer jp/ngt/rtm/modelpack/init/ProgressStateHolder
 
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/init/ModelPackLoadThread.jasm b/src/main/rtm/jp/ngt/rtm/modelpack/init/ModelPackLoadThread.jasm
index 85ac6d0..a8c04bf 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/init/ModelPackLoadThread.jasm
+++ b/src/main/rtm/jp/ngt/rtm/modelpack/init/ModelPackLoadThread.jasm
@@ -524,18 +524,18 @@ L_003e:
     invokestatic jp/ngt/ngtlib/io/NGTLog/startTimer ()V
 L_0041:
     .line 185
-    new jp/ngt/rtm/modelpack/init/ModelPackConstructThread
+    new com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread
     dup
     aload 0
     getfield jp/ngt/rtm/modelpack/init/ModelPackLoadThread/threadSide Lnet/minecraftforge/fml/relauncher/Side;
     aload 0
-    invokespecial jp/ngt/rtm/modelpack/init/ModelPackConstructThread/<init> (Lnet/minecraftforge/fml/relauncher/Side;Ljp/ngt/rtm/modelpack/init/ModelPackLoadThread;)V
+    invokespecial com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread/<init> (Lnet/minecraftforge/fml/relauncher/Side;Ljp/ngt/rtm/modelpack/init/ModelPackLoadThread;)V
     astore 1
 L_004a:
     .line 186
-    .var 1 is thread2 Ljp/ngt/rtm/modelpack/init/ModelPackConstructThread; from L_004a to L_006a
+    .var 1 is thread2 Lcom/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread; from L_004a to L_006a
     aload 1
-    invokevirtual jp/ngt/rtm/modelpack/init/ModelPackConstructThread/start ()V
+    invokevirtual com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread/start ()V
 L_004e:
     .line 187
     getstatic jp/ngt/rtm/modelpack/ModelPackManager/INSTANCE Ljp/ngt/rtm/modelpack/ModelPackManager;
@@ -545,10 +545,10 @@ L_0053:
     .line 188
     .stack
         locals Object jp/ngt/rtm/modelpack/init/ModelPackLoadThread
-        locals Object jp/ngt/rtm/modelpack/init/ModelPackConstructThread
+        locals Object com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread
     .end stack
     aload 1
-    invokevirtual jp/ngt/rtm/modelpack/init/ModelPackConstructThread/setFinish ()Z
+    invokevirtual com/anatawa12/fixRtm/rtm/modelpack/init/ExModelPackConstructThread/setFinish ()Z
     ifne L_005e
 L_0059:
     .line 190
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/state/ResourceState.jasm b/src/main/rtm/jp/ngt/rtm/modelpack/state/ResourceState.jasm
index ce341fe..c56013a 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/modelpack/state/ResourceState.jasm
+++ b/src/main/rtm/jp/ngt/rtm/modelpack/state/ResourceState.jasm
@@ -331,6 +331,9 @@ L_0009:
     aload 0
     getfield jp/ngt/rtm/modelpack/state/ResourceState/modelName Ljava/lang/String;
     invokevirtual jp/ngt/rtm/modelpack/ModelPackManager/getResourceSet (Ljp/ngt/rtm/modelpack/ResourceType;Ljava/lang/String;)Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
+    aload 0
+    getfield jp/ngt/rtm/modelpack/state/ResourceState/type Ljp/ngt/rtm/modelpack/ResourceType;
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/eraseNullForModelSet (Ljp/ngt/rtm/modelpack/modelset/ResourceSet;Ljp/ngt/rtm/modelpack/ResourceType;)Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
     putfield jp/ngt/rtm/modelpack/state/ResourceState/modelSet Ljp/ngt/rtm/modelpack/modelset/ResourceSet;
 L_0014:
     .line 118
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/rail/BlockMarker.jasm b/src/main/rtm/jp/ngt/rtm/rail/BlockMarker.jasm
index 8663826..e509f2d 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/rail/BlockMarker.jasm
+++ b/src/main/rtm/jp/ngt/rtm/rail/BlockMarker.jasm
@@ -572,6 +572,8 @@ L_0000:
     .var 4 is z I from L_0000 to L_0046
     .var 5 is player Lnet/minecraft/entity/player/EntityPlayer; from L_0000 to L_0046
     .var 6 is makeRail Z from L_0000 to L_0046
+    aload 5
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/BlockMarker_onMarkerActivated "(Lnet/minecraft/entity/player/EntityPlayer;)V"
     aload 0
     aload 5
     iload 6
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/rail/util/RailMaker.jasm b/src/main/rtm/jp/ngt/rtm/rail/util/RailMaker.jasm
index 61097af..41d3cbc 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/rail/util/RailMaker.jasm
+++ b/src/main/rtm/jp/ngt/rtm/rail/util/RailMaker.jasm
@@ -509,7 +509,7 @@ L_005a:
     getfield jp/ngt/rtm/rail/util/RailPosition/blockZ I
     invokestatic java/lang/Integer/valueOf (I)Ljava/lang/Integer;
     aastore
-    invokestatic jp/ngt/ngtlib/io/NGTLog/sendChatMessageToAll "(Ljava/lang/String;[Ljava/lang/Object;)V"
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/sendSwitchTypeError "(Ljava/lang/String;[Ljava/lang/Object;)V"
 L_0072:
     .line 101
     .stack use locals
diff --git a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/sound/MovingSoundMaker.jasm b/src/main/rtm/jp/ngt/rtm/sound/MovingSoundMaker.jasm
index 55fb34a..2b55a66 100644
--- a/mods/rtm.deobf.jar.src.processed/jp/ngt/rtm/sound/MovingSoundMaker.jasm
+++ b/src/main/rtm/jp/ngt/rtm/sound/MovingSoundMaker.jasm
@@ -317,6 +317,8 @@ L_0018:
     .line 115
     .var 3 is map Ljava/util/Map; signature "Ljava/util/Map<Ljava/lang/String;Ljp/ngt/rtm/sound/MovingSoundMaker\$SoundListDummy;>;" from L_0018 to L_009c
     aload 3
+    aload 0
+    invokestatic com/anatawa12/fixRtm/rtm/HooksKt/MovingSoundMaker_loadSoundJson_nullCheck (Ljava/util/Map;Ljava/lang/String;)Ljava/util/Map;
     invokeinterface java/util/Map/entrySet ()Ljava/util/Set;
     invokeinterface java/util/Set/iterator ()Ljava/util/Iterator;
     astore 4
