--- a/jp/ngt/rtm/modelpack/cfg/ResourceConfig.java
+++ b/jp/ngt/rtm/modelpack/cfg/ResourceConfig.java
@@ -5,10 +5,11 @@
 public abstract class ResourceConfig {
    public short version;
    public boolean useCustomColor;
    public String tags;
    public ResourceConfig.DMInitValue[] defaultValues;
+   public java.io.File file;
    /** @deprecated */
    @Deprecated
    public String defaultData;
 
    public abstract String getName();
@@ -28,14 +29,18 @@
       }
 
    }
 
    protected String fixSoundPath(String path) {
+      return fixSoundPath(path, null);
+   }
+
+   protected String fixSoundPath(String path, String defaults) {
       if (path != null && path.length() != 0) {
          return !path.contains(":") ? "rtm:" + path : path;
       } else {
-         return null;
+         return defaults;
       }
    }
 
    public class DMInitValue {
       public String type;
