--- a/jp/ngt/rtm/modelpack/state/ResourceState.java
+++ b/jp/ngt/rtm/modelpack/state/ResourceState.java
@@ -84,11 +84,11 @@
       this.setResourceName(this.type.defaultName);
    }
 
    public T getResourceSet() {
       if (this.modelSet == null || this.modelSet.isDummy()) {
-         this.modelSet = ModelPackManager.INSTANCE.getResourceSet(this.type, this.modelName);
+         this.modelSet = (T)com.anatawa12.fixRtm.rtm.HooksKt.eraseNullForModelSet(ModelPackManager.INSTANCE.getResourceSet(this.type, this.modelName), this.type);
          if (!this.modelSet.isDummy()) {
             this.modelSet.dataFormatter.initDataMap(this.dataMap);
          }
       }
 
