--- a/jp/ngt/rtm/modelpack/ModelPackManager.java
+++ b/jp/ngt/rtm/modelpack/ModelPackManager.java
@@ -46,19 +46,21 @@
 public final class ModelPackManager {
    public static final ModelPackManager INSTANCE = new ModelPackManager();
    private static final Pattern SC_INCLUDE = Pattern.compile("//include <(.+)>");
    private static final String[] ENCODING = new String[]{"UTF-8", "SJIS"};
    private final Map<String, ResourceType> typeMap = new HashMap<>(32);
-   private final Map<ResourceType, Map<String, ResourceSet>> allModelSetMap = new HashMap<>(256);
-   private final Map<ResourceType, Map<String, ResourceSet>> smpModelSetMap = new HashMap<>(256);
-   private final Map<String, ResourceSet> dummyMap = new HashMap<>(32);
-   public final List<ResourceSet> unconstructSets = new ArrayList<>(256);
+   public final Map<ResourceType, Map<String, ResourceSet>> allModelSetMap = new HashMap<>(256);
+   public final Map<ResourceType, Map<String, ResourceSet>> smpModelSetMap = new HashMap<>(256);
+   public final Map<String, ResourceSet> dummyMap = new HashMap<>(32);
+   public final java.util.concurrent.locks.ReadWriteLock modelSetMapLock = new java.util.concurrent.locks.ReentrantReadWriteLock();
+   public final com.anatawa12.fixRtm.rtm.modelpack.init.UnconstructSetsQueue unconstructSets = new com.anatawa12.fixRtm.rtm.modelpack.init.UnconstructSetsQueue();
    private final Map<String, IModelNGT> modelCache = new HashMap<>(128);
    private final Map<String, Map<String, ResourceLocation>> resourceCache = new HashMap<>(256);
    private final Map<String, String> scriptCache = new HashMap<>(64);
    public final List<File> fileCache = new ArrayList<>(1024);
    public boolean modelLoaded;
+   public volatile boolean modelConstructed;
    private static final String DEFAULT_DOMAIN = "minecraft";
 
    private ModelPackManager() {
    }
 
@@ -114,10 +116,11 @@
       for(File file2 : list2) {
          ++i;
          String s1 = file2.getName();
          par1.setBarValue(1, i, s1);
          RRSConfig rrsconfig = new RRSConfig(s1);
+         rrsconfig.file = file2;
          this.registerResourceSet(RTMResource.RRS, rrsconfig, "dummy_str");
       }
 
       TimeTableManager.INSTANCE.load();
       par1.loadFinished = true;
@@ -143,29 +146,43 @@
    public ResourceType getType(String name) {
       return this.typeMap.get(name);
    }
 
    public String registerResourceSet(ResourceType type, File file) throws IOException {
+      List<String> json = null;
       for(String s : ENCODING) {
          try {
-            String s1 = NGTText.readText(file, false, s);
-            ResourceConfig resourceconfig = (ResourceConfig)NGTJson.getObjectFromJson(s1, type.cfgClass);
-            return this.registerResourceSet(type, resourceconfig, s1);
+            json = NGTText.readText(file, s);
+            break;
          } catch (IOException ioexception) {
             ioexception.printStackTrace();
          }
       }
+      if (json == null)
+         throw new ModelPackException("Can't load json. please check encoding of json " +
+            "file and file is exists.", file.getAbsolutePath());
 
-      throw new ModelPackException("Can't load model", file.getAbsolutePath());
+      ResourceConfig resourceconfig = (ResourceConfig)NGTJson.getObjectFromJson(com.anatawa12.fixRtm.UtilsKt.joinLinesForJsonReading(json), type.cfgClass);
+      resourceconfig.file = file;
+      return this.registerResourceSet(type, resourceconfig, NGTText.append(json, false));
    }
 
    public String registerResourceSet(ResourceType type, ResourceConfig cfg, String origJson) {
       cfg.init();
       ResourceSet resourceset = this.getNewModelSet(type, new Class[]{type.cfgClass}, cfg);
       if (resourceset != null) {
-         NGTLog.debug("Registr resource : %s (%s)", new Object[]{cfg.getName(), type.name});
+         if (!com.anatawa12.fixRtm.asm.config.MainConfig.reduceConstructModelLog) {
+            NGTLog.debug("Registr resource : %s (%s)", new Object[]{cfg.getName(), type.name});
+         } else {
+            NGTLog.trace("Registr resource : %s (%s)", new Object[]{cfg.getName(), type.name});
+         }
+         modelSetMapLock.writeLock().lock();
+         try {
          this.allModelSetMap.get(type).put(cfg.getName(), resourceset);
+         } finally {
+            modelSetMapLock.writeLock().unlock();
+         }
          this.unconstructSets.add(resourceset);
          resourceset.md5 = EncryptedModel.getMD5(EncryptedModel.formatJson(origJson));
          return cfg.getName();
       } else {
          throw new ModelPackException("Failed to create ResourceSet", cfg.getName());
@@ -177,15 +194,16 @@
 
       for(Entry<ResourceType, Map<String, ResourceSet>> entry : this.allModelSetMap.entrySet()) {
          for(ResourceSet resourceset : entry.getValue().values()) {
             ResourceConfig resourceconfig = resourceset.getConfig();
             RTMCore.NETWORK_WRAPPER.sendTo(new PacketModelSet(i, (entry.getKey()).name, resourceconfig.getName()), player);
-            NGTLog.debug("[RTM] Send model to client : " + resourceconfig.getName());
+            NGTLog.trace("[RTM] Send model to client : " + resourceconfig.getName());
             ++i;
          }
       }
 
+      com.anatawa12.fixRtm.network.NetworkHandler.sendPacketEPM(new com.anatawa12.fixRtm.network.SentAllModels(), player);
    }
 
    public void addModelSetName(int count, String typeName, String name) {
       assert NGTUtil.isSMP() && !NGTUtil.isServer();
 
@@ -227,16 +245,13 @@
          resourcetype = type.parent;
       }
 
       boolean flag = NGTUtil.isSMP() && !NGTUtil.isServer();
       Map<ResourceType, Map<String, ResourceSet>> map = flag ? this.smpModelSetMap : this.allModelSetMap;
-      T t = map.get(resourcetype).get(name);
+      T t = (T)map.get(resourcetype).get(name);
       if (t == null) {
-         t = this.dummyMap.get(type.hasSubType ? type.subType : type.name);
-         if (t == null) {
-            throw new ModelPackException("Default model is not registered.", name);
-         }
+         return (T)com.anatawa12.fixRtm.DummyModelPackManager.getSet(type, name);
       }
 
       return t;
    }
 
