--- a/jp/ngt/ngtlib/io/NGTFileLoader.java
+++ b/jp/ngt/ngtlib/io/NGTFileLoader.java
@@ -38,11 +38,11 @@
       }
 
    }
 
    public static List<File> findFile(FileMatcher matcher) {
-      ScanResult scanresult = findFile(matcher);
+      ScanResult scanresult = findFile(new FileMatcher[] { matcher });
       return scanresult.asList();
    }
 
    public static ScanResult findFile(FileMatcher... matchers) {
       ScanResult scanresult = new ScanResult();
@@ -153,27 +153,18 @@
       if (MODS_DIR != null) {
          return MODS_DIR;
       } else {
          MODS_DIR = new ArrayList<>();
 
-         try {
-            File file1 = new File(Thread.currentThread().getContextClassLoader().getResource("").getPath());
-            if (!file1.getAbsolutePath().contains("mods")) {
-               MODS_DIR.add(file1);
-               NGTLog.debug("[NGTFL] Add mods dir : " + file1.getAbsolutePath());
-            }
-         } catch (NullPointerException nullpointerexception) {
-         }
-
-         File file2 = NGTCore.proxy.getMinecraftDirectory("mods");
-         String s = file2.getAbsolutePath();
-         if (s.contains(".") && !s.contains(".minecraft")) {
-            s = s.replace("\\.", "");
+         for (File jarOrDir : com.anatawa12.fixRtm.io.FIXFileLoader.INSTANCE.getModsOrJars()) {
+            if (jarOrDir.isDirectory())
+               MODS_DIR.add(jarOrDir);
+            else
+               MODS_DIR.add(jarOrDir.getParentFile());
          }
 
-         MODS_DIR.add(new File(s));
-         NGTLog.debug("[NGTFL] Add mods dir : " + s);
+         NGTLog.debug("[NGTFL] Add mods dir : " + MODS_DIR);
          return MODS_DIR;
       }
    }
 
    private static JFileChooser getCustomChooser(String title) {
@@ -231,23 +222,11 @@
          return null;
       }
    }
 
    public static InputStream getInputStream(ResourceLocation par1) throws IOException {
-      if (!NGTCore.proxy.isServer()) {
-         return NGTUtilClient.getMinecraft().getResourceManager().getResource(par1).getInputStream();
-      } else {
-         int i = par1.getPath().lastIndexOf("/");
-         String s = par1.getPath().substring(i + 1);
-         List<File> list = findFile((FileMatcher)((filex) -> filex.getName().equals(s)));
-         if (list.isEmpty()) {
-            throw new FileNotFoundException("On get stream : " + s);
-         } else {
-            File file1 = list.get(0);
-            return getInputStreamFromFile(file1);
-         }
-      }
+      return com.anatawa12.fixRtm.io.FIXFileLoader.INSTANCE.getInputStream(par1);
    }
 
    public static InputStream getInputStreamFromFile(File file) throws IOException {
       String s = getArchiveSuffix(file.getAbsolutePath());
       return (InputStream)(!s.isEmpty() ? getStreamFromArchive(file, s) : new FileInputStream(file));
