--- a/jp/ngt/ngtlib/io/NGTFileLoader.java
+++ b/jp/ngt/ngtlib/io/NGTFileLoader.java
@@ -38,11 +38,11 @@
       }
 
    }
 
    public static List<File> findFile(FileMatcher matcher) {
-      ScanResult scanresult = findFile(matcher);
+      ScanResult scanresult = findFile(new FileMatcher[] { matcher });
       return scanresult.asList();
    }
 
    public static ScanResult findFile(FileMatcher... matchers) {
       ScanResult scanresult = new ScanResult();
@@ -153,23 +153,25 @@
       if (MODS_DIR != null) {
          return MODS_DIR;
       } else {
          MODS_DIR = new ArrayList<>();
 
+         boolean dev = false;
          try {
             File file1 = new File(Thread.currentThread().getContextClassLoader().getResource("").getPath());
             if (!file1.getAbsolutePath().contains("mods")) {
                MODS_DIR.add(file1);
                NGTLog.debug("[NGTFL] Add mods dir : " + file1.getAbsolutePath());
+               dev = true;
             }
          } catch (NullPointerException nullpointerexception) {
          }
 
          File file2 = NGTCore.proxy.getMinecraftDirectory("mods");
          String s = file2.getAbsolutePath();
-         if (s.contains(".") && !s.contains(".minecraft")) {
-            s = s.replace("\\.", "");
+         if (dev) {
+            s = s.replaceAll("\\\\.\\\\mods$", "\\\\mods");
          }
 
          MODS_DIR.add(new File(s));
          NGTLog.debug("[NGTFL] Add mods dir : " + s);
          return MODS_DIR;
