--- a/jp/ngt/ngtlib/io/NGTText.java
+++ b/jp/ngt/ngtlib/io/NGTText.java
@@ -71,11 +71,12 @@
          } catch (UnsupportedEncodingException unsupportedencodingexception) {
             unsupportedencodingexception.printStackTrace();
             inputstreamreader = new InputStreamReader(is);
          }
       } else {
-         inputstreamreader = new InputStreamReader(is);
+         kotlin.Pair<java.nio.charset.Charset, InputStream> pair = com.anatawa12.fixRtm.ngtlib.renderer.model.PolygonModelCharsetDetector.INSTANCE.detectCharset(is, java.nio.charset.Charset.defaultCharset());
+         inputstreamreader = new InputStreamReader(pair.component2(), pair.component1());
       }
 
       BufferedReader bufferedreader = new BufferedReader(inputstreamreader);
       Stream<String> stream = bufferedreader.lines();
       stream.forEachOrdered((line) -> list.add(line));
