--- a/jp/ngt/rtm/item/ItemRail.java
+++ b/jp/ngt/rtm/item/ItemRail.java
@@ -156,12 +156,12 @@
 
    protected ResourceType getModelType(ItemStack itemStack) {
       return RTMResource.RAIL;
    }
 
-   public int getGuiId(ItemStack stack) {
-      return RTMCore.guiIdSelectItemModel;
+   @Override @SideOnly(Side.CLIENT) public net.minecraft.client.gui.GuiScreen newGuiScreen(ItemArgHolder holder) {
+      return newGuiSelectModel(holder);
    }
 
    protected ResourceState getNewState(ItemStack itemStack, ResourceType type) {
       return new ResourceStateRail(type, (Object)null);
    }
@@ -195,10 +195,11 @@
    }
 
    public static ItemStack copyItemFromRail(TileEntityLargeRailCore core) {
       ItemStack itemstack = getRailItem(core.getResourceState());
       RailPosition[] arailposition = core.getRailPositions();
+      if (arailposition == null) return ItemStack.EMPTY;
       setRPToItem(itemstack, arailposition);
       String s = core.getRailShapeName();
       itemstack.getTagCompound().setString("ShapeName", s);
       return itemstack;
    }
@@ -206,11 +207,11 @@
    private boolean placeRail(World world, BlockPos pos, ItemStack stack, EntityPlayer player) {
       List<RailPosition> list = getRPFromItem(stack);
       if (list.isEmpty()) {
          return false;
       } else {
-         int i = -BlockMarker.getFacing(player, false) * 2 + 4;
+         int i = BlockMarker.getMarkerDir(jp.ngt.rtm.RTMBlock.marker, com.anatawa12.fixRtm.rtm.rail.BlockMarker.getFaceMeta(player));
          RailPosition railposition = list.get(0);
          int j = i - railposition.direction;
          int k = railposition.blockX;
          int l = railposition.blockY;
          int i1 = railposition.blockZ;
