--- a/jp/ngt/rtm/item/ItemRail.java
+++ b/jp/ngt/rtm/item/ItemRail.java
@@ -156,12 +156,12 @@
 
    protected ResourceType getModelType(ItemStack itemStack) {
       return RTMResource.RAIL;
    }
 
-   public int getGuiId(ItemStack stack) {
-      return RTMCore.guiIdSelectItemModel;
+   public net.minecraft.client.gui.GuiScreen newGuiScreen(ItemArgHolder holder) {
+      return newGuiSelectModel(holder);
    }
 
    protected ResourceState getNewState(ItemStack itemStack, ResourceType type) {
       return new ResourceStateRail(type, (Object)null);
    }
@@ -195,10 +195,11 @@
    }
 
    public static ItemStack copyItemFromRail(TileEntityLargeRailCore core) {
       ItemStack itemstack = getRailItem(core.getResourceState());
       RailPosition[] arailposition = core.getRailPositions();
+      if (arailposition == null) return ItemStack.EMPTY;
       setRPToItem(itemstack, arailposition);
       String s = core.getRailShapeName();
       itemstack.getTagCompound().setString("ShapeName", s);
       return itemstack;
    }
