--- a/jp/ngt/rtm/item/ItemRail.java
+++ b/jp/ngt/rtm/item/ItemRail.java
@@ -196,10 +196,11 @@
    }
 
    public static ItemStack copyItemFromRail(TileEntityLargeRailCore core) {
       ItemStack itemstack = getRailItem(core.getResourceState());
       RailPosition[] arailposition = core.getRailPositions();
+      if (arailposition == null) return ItemStack.EMPTY;
       setRPToItem(itemstack, arailposition);
       String s = core.getRailShapeName();
       itemstack.getTagCompound().setString("ShapeName", s);
       return itemstack;
    }
