--- a/jp/ngt/rtm/item/ItemPaintTool.java
+++ b/jp/ngt/rtm/item/ItemPaintTool.java
@@ -72,11 +72,11 @@
 
    public void onUsingTick(ItemStack stack, EntityLivingBase living, int count) {
       if (living instanceof EntityPlayer) {
          EntityPlayer entityplayer = (EntityPlayer)living;
          float f = entityplayer.capabilities.isCreativeMode ? 5.0F : 4.5F;
-         RayTraceResult raytraceresult = entityplayer.rayTrace((double)f, 0.0F);
+         RayTraceResult raytraceresult = com.anatawa12.fixRtm.UtilsKt.rayTraceBothSide(entityplayer, (double)f, 0.0F);
          if (raytraceresult != null) {
             int i = raytraceresult.getBlockPos().getX();
             int j = raytraceresult.getBlockPos().getY();
             int k = raytraceresult.getBlockPos().getZ();
             float f1 = (float)raytraceresult.hitVec.x - (float)i;
