--- a/jp/ngt/rtm/event/RTMKeyHandlerClient.java
+++ b/jp/ngt/rtm/event/RTMKeyHandlerClient.java
@@ -51,25 +51,34 @@
    }
 
    public void onTickStart() {
       Minecraft minecraft = NGTUtilClient.getMinecraft();
       EntityPlayer entityplayer = minecraft.player;
-      if (Keyboard.isKeyDown(minecraft.gameSettings.keyBindJump.getKeyCode())) {
+      if (minecraft.gameSettings.keyBindJump.isKeyDown()) {
          if (entityplayer.isRiding() && entityplayer.getRidingEntity() instanceof EntityVehicle) {
             this.sendKeyToServer((byte)9, "");
          }
-      } else if (Keyboard.isKeyDown(minecraft.gameSettings.keyBindSneak.getKeyCode()) && entityplayer.isRiding() && entityplayer.getRidingEntity() instanceof EntityPlane && !entityplayer.getRidingEntity().onGround) {
+      } else if (minecraft.gameSettings.keyBindSneak.isKeyDown() && entityplayer.isRiding() && entityplayer.getRidingEntity() instanceof EntityPlane && !entityplayer.getRidingEntity().onGround) {
          this.sendKeyToServer((byte)10, "");
       }
 
    }
 
    public void onTickEnd() {
    }
 
    @SubscribeEvent
-   public void keyDown(KeyInputEvent event) {
+   public void onInputUpdateEvent(net.minecraftforge.client.event.InputUpdateEvent event) {
+      EntityPlayer player = event.getEntityPlayer();
+      Entity ridingEntity = player.getRidingEntity();
+      if (ridingEntity instanceof EntityPlane && ((EntityPlane)ridingEntity).disableUnmount()) {
+         event.getMovementInput().sneak = false;
+      }
+   }
+
+   @SubscribeEvent
+   public void keyDown(net.minecraftforge.fml.common.gameevent.InputEvent event) {
       Minecraft minecraft = NGTUtilClient.getMinecraft();
       EntityPlayer entityplayer = minecraft.player;
       Entity entity = entityplayer.getRidingEntity();
       if (minecraft.gameSettings.keyBindBack.isPressed()) {
          if (entityplayer.isRiding() && entity instanceof EntityTrainBase) {
@@ -79,13 +88,10 @@
          if (entityplayer.isRiding() && entity instanceof EntityTrainBase) {
             ((EntityTrainBase)entity).syncNotch(-1);
          }
       } else if (!minecraft.gameSettings.keyBindJump.isKeyDown()) {
          if (minecraft.gameSettings.keyBindSneak.isKeyDown()) {
-            if (entityplayer.isRiding() && entity instanceof EntityPlane && ((EntityPlane)entity).disableUnmount()) {
-               this.unpressKey(minecraft.gameSettings.keyBindSneak);
-            }
          } else if (KEY_HORN.isPressed()) {
             if (entityplayer.isRiding()) {
                if (entity instanceof EntityTrainBase) {
                   this.playSound(entityplayer, (byte)2);
                } else if (entity instanceof EntityArtillery) {
@@ -152,11 +158,11 @@
                   s = astring[i][1];
                   MacroRecorder.INSTANCE.recChime(player.world, astring[i][1]);
                }
             }
 
-            if (!s.isEmpty()) {
+            if (s != null && !s.isEmpty()) {
                this.sendKeyToServer(key, s);
             }
          }
       }
 
