--- a/jp/ngt/rtm/event/RTMKeyHandlerClient.java
+++ b/jp/ngt/rtm/event/RTMKeyHandlerClient.java
@@ -19,11 +19,11 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.text.TextComponentString;
 import net.minecraftforge.fml.client.registry.ClientRegistry;
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
-import net.minecraftforge.fml.common.gameevent.InputEvent.KeyInputEvent;
+import net.minecraftforge.fml.common.gameevent.InputEvent;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.lwjgl.input.Keyboard;
 
 @SideOnly(Side.CLIENT)
@@ -65,11 +65,11 @@
 
    public void onTickEnd() {
    }
 
    @SubscribeEvent
-   public void keyDown(KeyInputEvent event) {
+   public void keyDown(InputEvent event) {
       Minecraft minecraft = NGTUtilClient.getMinecraft();
       EntityPlayer entityplayer = minecraft.player;
       Entity entity = entityplayer.getRidingEntity();
       if (minecraft.gameSettings.keyBindBack.isPressed()) {
          if (entityplayer.isRiding() && entity instanceof EntityTrainBase) {
@@ -152,11 +152,11 @@
                   s = astring[i][1];
                   MacroRecorder.INSTANCE.recChime(player.world, astring[i][1]);
                }
             }
 
-            if (!s.isEmpty()) {
+            if (s != null && !s.isEmpty()) {
                this.sendKeyToServer(key, s);
             }
          }
       }
 
