--- a/jp/ngt/ngtlib/item/craft/RecipeManager.java
+++ b/jp/ngt/ngtlib/item/craft/RecipeManager.java
@@ -72,11 +72,11 @@
       for(int i1 = 0; i1 < j * k; ++i1) {
          char c0 = s.charAt(i1);
          if (map.containsKey(Character.valueOf(c0))) {
             aitemstack[i1] = ((ItemStack)map.get(Character.valueOf(c0))).copy();
          } else {
-            aitemstack[i1] = null;
+            aitemstack[i1] = ItemStack.EMPTY;
          }
       }
 
       ShapedRecipes55 shapedrecipes55 = new ShapedRecipes55(j, k, aitemstack, output);
       INSTANCE.addRecipeToManager(shapedrecipes55);
