--- a/jp/ngt/rtm/entity/train/util/Formation.java
+++ b/jp/ngt/rtm/entity/train/util/Formation.java
@@ -9,10 +9,11 @@
 import jp.ngt.rtm.network.PacketFormation;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
+import org.jetbrains.annotations.Nullable;
 
 public final class Formation {
    public final long id;
    public FormationEntry[] entries;
    private EntityTrainBase controlCar;
@@ -147,23 +148,25 @@
       this.entries = aformationentry;
    }
 
    public void connectTrain(EntityTrainBase par1, EntityTrainBase par2, int par3, int par4, Formation par5) {
       FormationEntry formationentry = this.getEntry(par1);
+      FormationEntry formationentry2 = par5.getEntry(par2);
+      if (formationentry2 == null) return;
       if (formationentry != null) {
          int i = 0;
          boolean flag = par3 == formationentry.dir;
          if (flag) {
             this.reverse();
          }
 
-         formationentry = par5.getEntry(par2);
-         flag = par4 == formationentry.dir;
+         flag = par4 == formationentry2.dir;
          if (!flag) {
             par5.reverse();
          }
 
+
          this.addAll(par5.entries);
          this.reallocation();
 
          for(FormationEntry formationentry1 : this.entries) {
             this.setSpeed(0.0F);
@@ -304,12 +307,13 @@
    private void sendPacket() {
       RTMCore.NETWORK_WRAPPER.sendToAll(new PacketFormation(this));
    }
 
    public boolean isFrontCar(EntityTrainBase train) {
-      EntityTrainBase entitytrainbase = this.direction == 0 ? this.entries[0].train : this.entries[this.entries.length - 1].train;
-      return train.equals(entitytrainbase);
+      FormationEntry entry = this.direction == 0 ? this.entries[0] : this.entries[this.entries.length - 1];
+      if (entry == null) return false;
+      return train.equals(entry.train);
    }
 
    public void updateTrainMovement() {
       EntityTrainBase entitytrainbase = null;
 
