--- a/jp/ngt/rtm/entity/train/util/TrainState.java
+++ b/jp/ngt/rtm/entity/train/util/TrainState.java
@@ -47,11 +47,27 @@
       return TrainState.TrainStateType.Door;
    }
 
    public static enum TrainStateType {
       Direction(0, "train_dir", 0, 1),
-      Notch(1, "notch", -8, 5),
+      Notch(1, "notch", -8, 5) {
+         @Override
+         protected byte getMin0(jp.ngt.rtm.entity.vehicle.EntityVehicleBase<?> train) {
+            jp.ngt.rtm.modelpack.cfg.VehicleBaseConfig config = train.getResourceState().getResourceSet().getConfig();
+            if (config instanceof jp.ngt.rtm.modelpack.cfg.TrainConfig)
+               return (byte) -(((jp.ngt.rtm.modelpack.cfg.TrainConfig)config).deccelerations.length - 1);
+            return super.getMin0(train);
+         }
+
+         @Override
+         protected byte getMax0(jp.ngt.rtm.entity.vehicle.EntityVehicleBase<?> train) {
+            jp.ngt.rtm.modelpack.cfg.VehicleBaseConfig config = train.getResourceState().getResourceSet().getConfig();
+            if (config instanceof jp.ngt.rtm.modelpack.cfg.TrainConfig)
+               return (byte) ((jp.ngt.rtm.modelpack.cfg.TrainConfig) config).accelerateions.length;
+            return super.getMin0(train);
+         }
+      },
       Signal(2, "signal", 0, 127),
       Door(4, "door", 0, 3),
       Light(5, "light", 0, 2),
       Pantograph(6, "pantograph", 0, 1),
       ChunkLoader(7, "chunk_loader", 0, 8),
@@ -79,7 +95,41 @@
             }
          }
 
          return Direction;
       }
+
+      public byte clap(byte value, jp.ngt.rtm.entity.vehicle.EntityVehicleBase<?> train) {
+         byte min = getMin(train);
+         byte max = getMax(train);
+         return value < min ? min
+            : value > max ? max
+            : value;
+      }
+
+      public byte getMax(jp.ngt.rtm.entity.vehicle.EntityVehicleBase<?> train) {
+         try {
+            return getMax0(train);
+         } catch (Throwable t) {
+            t.printStackTrace();
+            return max;
+         }
+      }
+
+      protected byte getMax0(jp.ngt.rtm.entity.vehicle.EntityVehicleBase<?> train) {
+         return max;
+      }
+
+      public byte getMin(jp.ngt.rtm.entity.vehicle.EntityVehicleBase<?> train) {
+         try {
+            return getMin0(train);
+         } catch (Throwable t) {
+            t.printStackTrace();
+            return min;
+         }
+      }
+
+      protected byte getMin0(jp.ngt.rtm.entity.vehicle.EntityVehicleBase<?> train) {
+         return min;
+      }
    }
 }
