--- a/jp/ngt/rtm/entity/npc/EntityNPC.java
+++ b/jp/ngt/rtm/entity/npc/EntityNPC.java
@@ -398,6 +398,12 @@
       return this.state;
    }
 
    public void setSwingingArms(boolean swingingArms) {
    }
+
+   @Override
+   public void addEntityCrashInfo(net.minecraft.crash.CrashReportCategory category) {
+      super.addEntityCrashInfo(category);
+      com.anatawa12.fixRtm.rtm.entity.npc.EntityNPCKt.addEntityCrashInfo(this, category);
+   }
 }
