--- a/jp/ngt/rtm/entity/npc/EntityMotorman.java
+++ b/jp/ngt/rtm/entity/npc/EntityMotorman.java
@@ -56,10 +56,17 @@
    public void entityInit() {
       super.entityInit();
       this.getDataManager().register(HELD_ITEM, new ItemStack(Items.APPLE));
    }
 
+   // force role to be motorman
+   public void onLivingUpdate() {
+      this.myRole = Role.MOTORMAN;
+      this.roleChanged = false;
+      super.onLivingUpdate();
+   }
+
    public void writeEntityToNBT(NBTTagCompound nbt) {
       super.writeEntityToNBT(nbt);
       if (this.getDiagram() != null) {
          NBTTagCompound nbttagcompound = new NBTTagCompound();
          ItemStack itemstack = this.getDiagram();
