--- a/jp/ngt/ngtlib/util/PermissionManager.java
+++ b/jp/ngt/ngtlib/util/PermissionManager.java
@@ -126,10 +126,16 @@
       }
 
    }
 
    public boolean hasPermission(ICommandSender player, String category) {
+      if (com.anatawa12.fixRtm.asm.config.MainConfig.addNegativePermissionEnabled) {
+         if (this.getPlayerList("negative.".concat(category)).contains(player.getName())) {
+            NGTLog.sendChatMessageToAll("%s need permission (%s).", player.getName(), category);
+            return false;
+         }
+      }
       if (this.isOp(player)) {
          return true;
       } else {
          List<String> list = this.getPlayerList(category);
          if (!list.contains(player.getName()) && !list.contains(ALL)) {
@@ -142,10 +148,12 @@
    }
 
    public boolean isOp(ICommandSender player) {
       if (!DEBUG_MODE && !NGTUtil.isSMP()) {
          return true;
+      } else if (player == null) {
+         return true;
       } else {
          if (player instanceof EntityPlayerMP) {
             String[] astring = ((EntityPlayerMP)player).server.getPlayerList().getOppedPlayerNames();
 
             for(String s : astring) {
