--- a/jp/ngt/rtm/network/PacketVehicleMovement.java
+++ b/jp/ngt/rtm/network/PacketVehicleMovement.java
@@ -74,16 +74,20 @@
       this.vehicleRoll = buffer.readFloat();
       this.vehicleSpeed = buffer.readFloat();
    }
 
    public IMessage onMessage(PacketVehicleMovement message, MessageContext ctx) {
+      com.anatawa12.fixRtm.ThreadUtil.runOnClientThread(() -> this.doMessage(message));
+      return null;
+   }
+
+   private void doMessage(PacketVehicleMovement message) {
       World world = NGTUtil.getClientWorld();
       if (world == null) {
-         return null;
+         return;
       } else if (message.vehicleY < 0) {
          this.deleteEntity(world, message.entityId);
-         return null;
       } else {
          Entity entity = world.getEntityByID(message.entityId);
          if (entity != null && !entity.isDead) {
             double d0 = (double)message.vehicleX * 0.03125D;
             double d1 = (double)message.vehicleY * 0.03125D;
@@ -98,12 +102,10 @@
                ((EntityBogie)entity).setRoll(message.vehicleRoll);
             }
          } else {
             NGTLog.debug("[PVM] Entity is null or dead %d", new Object[]{message.entityId});
          }
-
-         return null;
       }
    }
 
    private void deleteEntity(World world, int id) {
       Entity entity = world.getEntityByID(id);
