--- a/jp/ngt/rtm/sound/MovingSoundTrain.java
+++ b/jp/ngt/rtm/sound/MovingSoundTrain.java
@@ -17,11 +17,11 @@
       super.update();
       if (this.changePitch) {
          EntityTrainBase entitytrainbase = (EntityTrainBase)this.entity;
          ModelSetTrain modelsettrain = entitytrainbase.getResourceState().getResourceSet();
          float f = (modelsettrain.getConfig()).maxSpeed[0];
-         float f1 = (entitytrainbase.getSpeed() - f) / ((modelsettrain.getConfig()).maxSpeed[4] - f) + 1.0F;
+         float f1 = (Math.abs(entitytrainbase.getSpeed()) - f) / ((modelsettrain.getConfig()).maxSpeed[4] - f) + 1.0F;
          this.pitch = f1;
       }
 
    }
 }
