--- a/jp/ngt/rtm/sound/SoundUpdaterTrain.java
+++ b/jp/ngt/rtm/sound/SoundUpdaterTrain.java
@@ -24,11 +24,11 @@
    public void update() {
       EntityTrainBase entitytrainbase = this.vehicle;
       EntityBogie entitybogie = entitytrainbase.getBogie(entitytrainbase.getTrainDirection());
       if (entitybogie != null) {
          int i = entitytrainbase.getSignal();
-         if (entitytrainbase.getSpeed() > 0.0F) {
+         if (Math.abs(entitytrainbase.getSpeed()) > 0.0F) {
             switch(i) {
             case -1:
                if (this.currentSignal != -1) {
                   if (this.atsSound[1] != null) {
                      this.atsSound[1].stop();
@@ -45,13 +45,14 @@
 
                   if (this.atsSound[1] != null) {
                      this.atsSound[1].stop();
                   }
 
-                  this.atsSound[0] = MovingSoundMaker.create(entitybogie, "rtm:sounds/train/ats.ogg", true);
+                  ModelSetVehicleBase<TrainConfig> modelSet = this.getEntity().getResourceState().getResourceSet();
+                  this.atsSound[0] = MovingSoundMaker.create(entitybogie, modelSet.getConfig().sound_ATSChime, true);
                   this.soundHandler.playSound(this.atsSound[0]);
-                  this.atsSound[1] = MovingSoundMaker.create(entitybogie, "rtm:sounds/train/ats_bell.ogg", true);
+                  this.atsSound[1] = MovingSoundMaker.create(entitybogie, modelSet.getConfig().sound_ATSBell, true);
                   this.soundHandler.playSound(this.atsSound[1]);
                   this.currentSignal = 1;
                }
             }
          } else if (i != -1 && this.currentSignal != 0) {
@@ -69,11 +70,11 @@
          super.update();
       }
    }
 
    protected String getSound(ModelSetVehicleBase modelset) {
-      float f = this.vehicle.getSpeed();
+      float f = Math.abs(this.vehicle.getSpeed());
       if (f > 0.0F) {
          float f1 = EnumNotch.getAcceleration(this.getNotch(), f);
          if (f < ((TrainConfig)modelset.getConfig()).maxSpeed[0]) {
             return f1 > 0.0F ? modelset.getConfig().sound_S_A : modelset.getConfig().sound_D_S;
          } else {
@@ -84,14 +85,18 @@
       }
    }
 
    protected boolean changePitch() {
       ModelSetTrain modelsettrain = this.vehicle.getResourceState().getResourceSet();
-      float f = this.vehicle.getSpeed();
+      float f = Math.abs(this.vehicle.getSpeed());
       return f > 0.0F ? !(f < (modelsettrain.getConfig()).maxSpeed[0]) : false;
    }
 
+   public float getSpeed() {
+      return Math.abs(this.vehicle.getSpeed()) * 72.0F;
+   }
+
    public float getMaxSpeed() {
       return (this.vehicle.getResourceState().getResourceSet().getConfig()).maxSpeed[4] * 72.0F;
    }
 
    public int getNotch() {
