--- a/jp/ngt/ngtlib/network/PacketNBT.java
+++ b/jp/ngt/ngtlib/network/PacketNBT.java
@@ -136,9 +136,16 @@
 
    public static void sendToClient(Entity entity) {
       NGTCore.NETWORK_WRAPPER.sendToAll(new PacketNBT(entity, true));
    }
 
+   public static void sendToClient(Entity entity, EntityPlayerMP player) {
+      NGTCore.NETWORK_WRAPPER.sendTo(new PacketNBT(entity, true), player);
+   }
+
+   /** @deprecated because it's slow. */
+   @Deprecated
    public static void sendToClient(TileEntity entity) {
+      com.anatawa12.fixRtm.Deprecation.found("PacketNBT#sendToClient");
       NGTCore.NETWORK_WRAPPER.sendToAll(new PacketNBT(entity, true));
    }
 }
