--- a/jp/ngt/ngtlib/network/PacketNBT.java
+++ b/jp/ngt/ngtlib/network/PacketNBT.java
@@ -86,10 +86,12 @@
             TileEntity tileentity = this.getTileEntity(world);
             if (tileentity != null) {
                tileentity.readFromNBT(this.nbtData);
                if (!world.isRemote) {
                   tileentity.markDirty();
+                  // markDirty is not enough to sync data
+                  jp.ngt.ngtlib.block.BlockUtil.markBlockForUpdate(tileentity.getWorld(), tileentity.getPos());
                }
 
                return true;
             }
          } else if (b0 == 2) {
@@ -136,9 +138,16 @@
 
    public static void sendToClient(Entity entity) {
       NGTCore.NETWORK_WRAPPER.sendToAll(new PacketNBT(entity, true));
    }
 
+   public static void sendToClient(Entity entity, EntityPlayerMP player) {
+      NGTCore.NETWORK_WRAPPER.sendTo(new PacketNBT(entity, true), player);
+   }
+
+   /** @deprecated because it's slow. */
+   @Deprecated
    public static void sendToClient(TileEntity entity) {
+      com.anatawa12.fixRtm.Deprecation.found("PacketNBT#sendToClient");
       NGTCore.NETWORK_WRAPPER.sendToAll(new PacketNBT(entity, true));
    }
 }
