--- a/jp/ngt/rtm/RTMEntity.java
+++ b/jp/ngt/rtm/RTMEntity.java
@@ -86,12 +86,12 @@
       NGTUtilClient.registerEntityRender(EntityVehicle.class, RenderVehicleBase.class);
       NGTUtilClient.registerEntityRender(EntityTrainBase.class, RenderVehicleBase.class);
       NGTUtilClient.registerEntityRender(WeatherEffectDummy.class, RenderWeatherEffectDummy.class);
       NGTUtilClient.registerEntityRender(EntityBogie.class, RenderBogie.class);
       NGTUtilClient.registerEntityRender(EntityFloor.class, RenderSeat.class);
-      NGTUtilClient.registerEntityRender(EntityATC.class, RenderEntityInstalledObject.class);
-      NGTUtilClient.registerEntityRender(EntityTrainDetector.class, RenderEntityInstalledObject.class);
+      NGTUtilClient.registerEntityRender(EntityATC.class, com.anatawa12.fixRtm.rtm.entity.RenderEntityElectricalWiring.class);
+      NGTUtilClient.registerEntityRender(EntityTrainDetector.class, com.anatawa12.fixRtm.rtm.entity.RenderEntityElectricalWiring.class);
       NGTUtilClient.registerEntityRender(EntityBumpingPost.class, RenderEntityInstalledObject.class);
       NGTUtilClient.registerEntityRender(EntityContainer.class, RenderContainer.class);
       NGTUtilClient.registerEntityRender(EntityArtillery.class, RenderArtillery.class);
       NGTUtilClient.registerEntityRender(EntityBullet.class, RenderBullet.class);
       NGTUtilClient.registerEntityRender(EntityTie.class, RenderTie.class);
