--- a/jp/ngt/rtm/RTMItem.java
+++ b/jp/ngt/rtm/RTMItem.java
@@ -134,11 +134,11 @@
       registerItemModels(installedObject, "istl_obj_", ItemInstalledObject.IstlObjType.values());
       registerItemModel(itemtrain, 0, "item_train_0");
       registerItemModel(itemtrain, 1, "item_train_1");
       registerItemModel(itemtrain, 2, "item_train_2");
       registerItemModel(itemtrain, 3, "item_train_3");
-      registerItemModel(itemtrain, 127, "item_train_127");
+      registerItemModel(itemtrain, 127, "item_train_fixrtm_test");
       registerItemModel(itemMotorman, 0, "item_npc_0");
       registerItemModel(itemMotorman, 1, "item_npc_1");
       registerItemModel(itemCargo, 0, "cargo_0");
       registerItemModel(itemCargo, 1, "cargo_1");
       registerItemModel(itemCargo, 2, "cargo_2");
