--- a/jp/ngt/rtm/RTMSound.java
+++ b/jp/ngt/rtm/RTMSound.java
@@ -36,10 +36,11 @@
    };
 
    public static void init() {
    }
 
+   @net.minecraftforge.fml.relauncher.SideOnly(net.minecraftforge.fml.relauncher.Side.CLIENT)
    public static void registerSoundDomains() {
       NGTLog.debug("[RTMSound] Register sound domains");
       SimpleReloadableResourceManager simplereloadableresourcemanager = (SimpleReloadableResourceManager)NGTUtilClient.getMinecraft().getResourceManager();
       Set<String> set = simplereloadableresourcemanager.getResourceDomains();
       List<File> list = new ArrayList<>();
@@ -53,21 +54,20 @@
                list.add(file2);
                set.add(file2.getName());
             }
 
             String s = file1.getAbsolutePath();
-            String s1 = s.substring(s.indexOf(file2.getName()) + file2.getName().length() + 1);
-            s1 = s1.replace("\\", "/");
+            s = s.replace("\\", "/");
+            String find = "/assets/" + file2.getName() + "/";
+            String s1 = s.substring(s.indexOf(find) + find.length());
             ALL_OGG_FILES.add(file2.getName() + ":" + s1);
          }
       }
 
-      Map map = getDomainResourceManagers(simplereloadableresourcemanager);
-
       for(File file3 : list) {
          RTMResourceManager rtmresourcemanager = new RTMResourceManager(getMetadataSerializer(simplereloadableresourcemanager), file3);
-         map.put(file3.getName(), rtmresourcemanager);
+         simplereloadableresourcemanager.domainResourceManagers.put(file3.getName(), rtmresourcemanager);
          NGTLog.debug("[RTMSound] Add new domain : " + file3.getName());
       }
 
    }
 
