--- a/jp/ngt/rtm/RTMSound.java
+++ b/jp/ngt/rtm/RTMSound.java
@@ -53,20 +53,23 @@
                list.add(file2);
                set.add(file2.getName());
             }
 
             String s = file1.getAbsolutePath();
-            String s1 = s.substring(s.indexOf(file2.getName()) + file2.getName().length() + 1);
-            s1 = s1.replace("\\", "/");
+            s = s.replace("\\", "/");
+            String find = "/assets/" + file2.getName() + "/";
+            String s1 = s.substring(s.indexOf(find) + find.length());
             ALL_OGG_FILES.add(file2.getName() + ":" + s1);
          }
       }
 
       Map map = getDomainResourceManagers(simplereloadableresourcemanager);
 
       for(File file3 : list) {
-         RTMResourceManager rtmresourcemanager = new RTMResourceManager(getMetadataSerializer(simplereloadableresourcemanager), file3);
+         net.minecraft.client.resources.FallbackResourceManager rtmresourcemanager =
+                 new net.minecraft.client.resources.FallbackResourceManager(getMetadataSerializer(simplereloadableresourcemanager));
+         rtmresourcemanager.addResourcePack(com.anatawa12.fixRtm.rtm.modelpack.FixRTMResourcePack.createPack(file3.getName(), file3));
          map.put(file3.getName(), rtmresourcemanager);
          NGTLog.debug("[RTMSound] Add new domain : " + file3.getName());
       }
 
    }
