--- a/jp/ngt/rtm/electric/BlockInsulator.java
+++ b/jp/ngt/rtm/electric/BlockInsulator.java
@@ -11,6 +11,11 @@
    }
 
    protected ItemStack getItem(int damage) {
       return new ItemStack(RTMItem.installedObject, 1, 3);
    }
+
+   @Override
+   protected jp.ngt.rtm.item.ItemInstalledObject.IstlObjType istlObjType(int damage) {
+      return jp.ngt.rtm.item.ItemInstalledObject.IstlObjType.INSULATOR;
+   }
 }
