--- a/jp/ngt/rtm/electric/BlockSignal.java
+++ b/jp/ngt/rtm/electric/BlockSignal.java
@@ -24,14 +24,21 @@
    public boolean onBlockActivated(BlockArgHolder holder, float hitX, float hitY, float hitZ) {
       if (holder.getWorld().isRemote) {
          int i = holder.getBlockPos().getX();
          int j = holder.getBlockPos().getY();
          int k = holder.getBlockPos().getZ();
-         holder.getPlayer().openGui(RTMCore.instance, RTMCore.guiIdSelectTileEntityModel, holder.getWorld(), i, j, k);
+         if (holder.getPlayer().inventory.getCurrentItem().getItem() == jp.ngt.rtm.RTMItem.crowbar) {
+            com.anatawa12.fixRtm.UtilsKt.openGui(holder.getPlayer(), com.anatawa12.fixRtm.gui.GuiId.ChangeOffset, holder.getWorld(), i, j, k);
+            return true;
+         }
+         if (holder.getPlayer().isSneaking()) {
+            holder.getPlayer().openGui(RTMCore.instance, RTMCore.guiIdSelectTileEntityModel, holder.getWorld(), i, j, k);
+            return true;
+         }
       }
 
-      return true;
+      return false;
    }
 
    public boolean removedByPlayer(BlockArgHolder holder, boolean willHarvest) {
       World world = holder.getWorld();
       BlockPos blockpos = holder.getBlockPos();
@@ -51,6 +58,11 @@
    }
 
    public boolean canConnect(World world, int x, int y, int z) {
       return true;
    }
+
+   @Override
+   public net.minecraft.item.ItemStack getPickBlock(net.minecraft.block.state.IBlockState state, net.minecraft.util.math.RayTraceResult target, World world, net.minecraft.util.math.BlockPos pos, net.minecraft.entity.player.EntityPlayer player) {
+      return com.anatawa12.fixRtm.rtm.block.BlockOrnamentMain.getPickBlock(world, pos, jp.ngt.rtm.item.ItemInstalledObject.IstlObjType.FLUORESCENT);
+   }
 }
