--- a/jp/ngt/rtm/electric/BlockInsulator.java
+++ b/jp/ngt/rtm/electric/BlockInsulator.java
@@ -1,16 +1,24 @@
 package jp.ngt.rtm.electric;
 
 import jp.ngt.rtm.RTMItem;
+import jp.ngt.rtm.item.ItemInstalledObject;
+import net.minecraft.block.state.IBlockState;
 import net.minecraft.item.ItemStack;
 import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
 public class BlockInsulator extends BlockConnector {
    public TileEntity createNewTileEntity(World world, int par2) {
       return new TileEntityInsulator();
    }
 
    protected ItemStack getItem(int damage) {
       return new ItemStack(RTMItem.installedObject, 1, 3);
    }
+
+   @Override
+   protected ItemInstalledObject.IstlObjType istlObjType(int damage) {
+      return jp.ngt.rtm.item.ItemInstalledObject.IstlObjType.INSULATOR;
+   }
 }
