--- a/jp/ngt/rtm/electric/RenderElectricalWiring.java
+++ b/jp/ngt/rtm/electric/RenderElectricalWiring.java
@@ -37,10 +37,12 @@
    }
 
    protected void renderConnector(TileEntityConnectorBase tileEntity, double par2, double par4, double par6, float par8, int pass) {
       GL11.glPushMatrix();
       GL11.glTranslatef((float)par2 + 0.5F, (float)par4 + 0.5F, (float)par6 + 0.5F);
+      GL11.glTranslatef(tileEntity.getOffsetX(), tileEntity.getOffsetY(), tileEntity.getOffsetZ());
+      GL11.glRotatef(tileEntity.getRotation(), 0.0F, 1.0F, 0.0F);
       ModelSetConnector modelsetconnector = tileEntity.getResourceState().getResourceSet();
       ConnectorConfig connectorconfig = modelsetconnector.getConfig();
       int i = tileEntity.getBlockMetadata() % 6;
       switch(i) {
       case 0:
