--- a/jp/ngt/rtm/electric/TileEntitySignal.java
+++ b/jp/ngt/rtm/electric/TileEntitySignal.java
@@ -167,6 +167,12 @@
    }
 
    public ResourceStateWithBlock<ModelSetSignal> getResourceState() {
       return this.state;
    }
+
+   @Override
+   public void addInfoToCrashReport(net.minecraft.crash.CrashReportCategory reportCategory) {
+      super.addInfoToCrashReport(reportCategory);
+      com.anatawa12.fixRtm.rtm.electric.TileEntitySignalKt.addInfoToCrashReport(this, reportCategory);
+   }
 }
