--- a/jp/ngt/ngtlib/block/TileEntityPlaceable.java
+++ b/jp/ngt/ngtlib/block/TileEntityPlaceable.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.nbt.NBTTagCompound;
 
 public abstract class TileEntityPlaceable extends TileEntityCustom {
    private float rotation;
+   private float offsetX, offsetY, offsetZ;
 
    public void readFromNBT(NBTTagCompound nbt) {
       super.readFromNBT(nbt);
       this.setRotation(nbt.getFloat("Yaw"), false);
    }
@@ -33,6 +34,28 @@
 
    public void setRotation(EntityPlayer player, float rotationInterval, boolean synch) {
       int i = NGTMath.floor(NGTMath.normalizeAngle((double)(-player.rotationYaw) + 180.0D + (double)rotationInterval / 2.0D) / (double)rotationInterval);
       this.setRotation((float)i * rotationInterval, synch);
    }
+
+   public float getOffsetX() {
+      return offsetX;
+   }
+
+   public float getOffsetY() {
+      return offsetY;
+   }
+
+   public float getOffsetZ() {
+      return offsetZ;
+   }
+
+   public void setOffset(float offsetX, float offsetY, float offsetZ, boolean sync) {
+      this.offsetX = offsetX;
+      this.offsetY = offsetY;
+      this.offsetZ = offsetZ;
+      if (sync) {
+         this.getUpdatePacket();
+         this.markDirty();
+      }
+   }
 }
