--- a/jp/ngt/ngtlib/block/TileEntityCustom.java
+++ b/jp/ngt/ngtlib/block/TileEntityCustom.java
@@ -30,17 +30,10 @@
    public int getZ() {
       return this.getPos().getZ();
    }
 
    public SPacketUpdateTileEntity getUpdatePacket() {
-      World world = this.getWorld();
-      TickProcessQueue.getInstance(Side.SERVER).add(new TickProcessEntry() {
-         public boolean process(World world) {
-            TileEntityCustom.this.sendPacket();
-            return true;
-         }
-      }, 40);
       return new SPacketUpdateTileEntity(this.pos, -1, this.getUpdateTag());
    }
 
    public NBTTagCompound getUpdateTag() {
       return this.writeToNBT(new NBTTagCompound());
@@ -50,11 +43,12 @@
       this.readFromNBT(pkt.getNbtCompound());
    }
 
    protected void sendPacket() {
       if (this.world == null || !this.world.isRemote) {
-         PacketNBT.sendToClient(this);
+         net.minecraft.block.state.IBlockState state = getWorld().getBlockState(getPos());
+         getWorld().notifyBlockUpdate(getPos(), state, state, 0);
       }
 
    }
 
    @SideOnly(Side.CLIENT)
