--- a/jp/ngt/rtm/render/RTMRenderers.java
+++ b/jp/ngt/rtm/render/RTMRenderers.java
@@ -9,14 +9,19 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 @SideOnly(Side.CLIENT)
 public final class RTMRenderers {
    public static <R extends PartsRenderer> R getRendererWithScript(ResourceLocation par1, String... args) {
+      com.anatawa12.fixRtm.asm.Preprocessor.ifDisabled("useOurScripting");
       String s = ModelPackManager.INSTANCE.getScript(par1.getPath());
       ScriptEngine scriptengine = ScriptUtil.doScript(s);
+      com.anatawa12.fixRtm.asm.Preprocessor.ifEnabled("useOurScripting");
+      s = null;
+      scriptengine = com.anatawa12.fixRtm.scripting.FIXScriptUtil.getScriptAndDoScript(par1.toString());
+      com.anatawa12.fixRtm.asm.Preprocessor.whatever("useOurScripting");
       String s1 = (String)ScriptUtil.getScriptField(scriptengine, "renderClass");
-      R r = getRenderer(s1, args);
+      R r = (R)getRenderer(s1, args);
       r.setScript(scriptengine, par1);
       scriptengine.put("renderer", r);
       return r;
    }
 
