--- a/jp/ngt/rtm/block/tileentity/TileEntityCrossingGate.java
+++ b/jp/ngt/rtm/block/tileentity/TileEntityCrossingGate.java
@@ -66,13 +66,14 @@
 
    @SideOnly(Side.CLIENT)
    public AxisAlignedBB getRenderBoundingBox() {
       float[] afloat = ((MachineConfig)((ModelSetMachine)this.getResourceState().getResourceSet()).getConfig()).renderAABB;
       BlockPos blockpos = this.getPos();
-      AxisAlignedBB axisalignedbb = new AxisAlignedBB((double)((float)blockpos.getX() + afloat[0]), (double)((float)blockpos.getY() + afloat[1]), (double)((float)blockpos.getZ() + afloat[2]), (double)((float)blockpos.getX() + afloat[3]), (double)((float)blockpos.getY() + afloat[4]), (double)((float)blockpos.getZ() + afloat[5]));
+      AxisAlignedBB axisalignedbb = new AxisAlignedBB((double)((float)blockpos.getX() + afloat[0]), (double)((float)blockpos.getY() + afloat[1]), (double)((float)blockpos.getZ() + afloat[2]), (double)((float)blockpos.getX() + afloat[3]), (double)((float)blockpos.getY() + afloat[4]), (double)((float)blockpos.getZ() + afloat[5]))
+         .offset(getOffsetX(), getOffsetY(), getOffsetZ());
       return axisalignedbb;
    }
 
-   protected ResourceType getSubType() {
+   public ResourceType getSubType() {
       return RTMResource.MACHINE_GATE;
    }
 }
