--- a/jp/ngt/rtm/block/tileentity/TileEntityPoint.java
+++ b/jp/ngt/rtm/block/tileentity/TileEntityPoint.java
@@ -44,12 +44,13 @@
       this.markDirty();
    }
 
    @SideOnly(Side.CLIENT)
    public AxisAlignedBB getRenderBoundingBox() {
-      return new AxisAlignedBB(this.getPos().add(-1, 0, -1), this.getPos().add(2, 1, 2));
+      return new AxisAlignedBB(this.getPos().add(-1, 0, -1), this.getPos().add(2, 1, 2))
+         .offset(getOffsetX(), getOffsetY(), getOffsetZ());
    }
 
-   protected ResourceType getSubType() {
+   public ResourceType getSubType() {
       return RTMResource.MACHINE_POINT;
    }
 }
