--- a/jp/ngt/rtm/block/tileentity/TileEntityOrnament.java
+++ b/jp/ngt/rtm/block/tileentity/TileEntityOrnament.java
@@ -11,11 +11,11 @@
 import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-public abstract class TileEntityOrnament extends TileEntityCustom implements IResourceSelector {
+public abstract class TileEntityOrnament extends jp.ngt.ngtlib.block.TileEntityPlaceable implements IResourceSelector {
    private ResourceState<ModelSetOrnament> state = new ResourceState<>(this.getSubType(), this);
    private byte attachedSide;
    private float randomScale;
 
    public void readFromNBT(NBTTagCompound nbt) {
@@ -83,6 +83,12 @@
    public boolean closeGui(ResourceState par1) {
       return true;
    }
 
    protected abstract ResourceType getSubType();
+
+   @Override
+   public void addInfoToCrashReport(net.minecraft.crash.CrashReportCategory reportCategory) {
+      super.addInfoToCrashReport(reportCategory);
+      com.anatawa12.fixRtm.rtm.block.tileentity.TileEntityOrnamentKt.addInfoToCrashReport(this, reportCategory);
+   }
 }
