--- a/jp/ngt/rtm/block/tileentity/TileEntityFluorescent.java
+++ b/jp/ngt/rtm/block/tileentity/TileEntityFluorescent.java
@@ -37,9 +37,18 @@
          }
       }
 
    }
 
-   protected ResourceType getSubType() {
+   public ResourceType getSubType() {
       return RTMResource.ORNAMENT_LAMP;
    }
+
+   @Override
+   public void setRotation(net.minecraft.entity.player.EntityPlayer player, float rotationInterval, boolean synch) {
+      int yaw = net.minecraft.util.math.MathHelper.floor(jp.ngt.ngtlib.math.NGTMath.normalizeAngle(-player.rotationYaw + 180.0D + (rotationInterval / 2.0D)) / (double) rotationInterval);
+      if (this.dirF >= 4) {
+         yaw -= 90;
+      }
+      this.setRotation((float) yaw * rotationInterval, synch);
+   }
 }
