--- a/jp/ngt/rtm/block/tileentity/TileEntityMachineBase.java
+++ b/jp/ngt/rtm/block/tileentity/TileEntityMachineBase.java
@@ -111,7 +111,13 @@
 
    public boolean closeGui(ResourceState par1) {
       return true;
    }
 
-   protected abstract ResourceType getSubType();
+   public abstract ResourceType getSubType();
+
+   @Override
+   public void addInfoToCrashReport(net.minecraft.crash.CrashReportCategory reportCategory) {
+      super.addInfoToCrashReport(reportCategory);
+      com.anatawa12.fixRtm.rtm.block.tileentity.TileEntityMachineBaseKt.addInfoToCrashReport(this, reportCategory);
+   }
 }
