--- a/jp/ngt/rtm/block/tileentity/TileEntityMachineBase.java
+++ b/jp/ngt/rtm/block/tileentity/TileEntityMachineBase.java
@@ -85,11 +85,12 @@
       return pass >= 0;
    }
 
    @SideOnly(Side.CLIENT)
    public AxisAlignedBB getRenderBoundingBox() {
-      return new AxisAlignedBB(this.getPos(), this.getPos().add(1, 1, 1));
+      return new AxisAlignedBB(this.getPos(), this.getPos().add(1, 1, 1))
+         .offset(getOffsetX(), getOffsetY(), getOffsetZ());
    }
 
    public ResourceState<ModelSetMachine> getResourceState() {
       return this.state;
    }
@@ -109,7 +110,13 @@
 
    public boolean closeGui(ResourceState par1) {
       return true;
    }
 
-   protected abstract ResourceType getSubType();
+   public abstract ResourceType getSubType();
+
+   @Override
+   public void addInfoToCrashReport(net.minecraft.crash.CrashReportCategory reportCategory) {
+      super.addInfoToCrashReport(reportCategory);
+      com.anatawa12.fixRtm.rtm.block.tileentity.TileEntityMachineBaseKt.addInfoToCrashReport(this, reportCategory);
+   }
 }
