--- a/jp/ngt/rtm/block/BlockRailroadSign.java
+++ b/jp/ngt/rtm/block/BlockRailroadSign.java
@@ -38,15 +38,20 @@
          int i = holder.getBlockPos().getX();
          int j = holder.getBlockPos().getY();
          int k = holder.getBlockPos().getZ();
          EntityPlayer entityplayer = holder.getPlayer();
          RTMCore rtmcore = RTMCore.instance;
+         if (holder.getPlayer().inventory.getCurrentItem().getItem() == RTMItem.crowbar) {
+            com.anatawa12.fixRtm.UtilsKt.openGui(holder.getPlayer(), com.anatawa12.fixRtm.gui.GuiId.ChangeOffset, holder.getWorld(), i, j, k);
+            return true;
+         }
          entityplayer.openGui(RTMCore.instance, RTMCore.guiIdSelectTileEntityTexture, holder.getWorld(), i, j, k);
       }
 
       return true;
    }
 
-   protected ItemStack getItem(int damage) {
-      return new ItemStack(RTMItem.installedObject, 1, ItemInstalledObject.IstlObjType.RAILLOAD_SIGN.id);
+   @Override
+   public ItemStack getPickBlock(IBlockState state, net.minecraft.util.math.RayTraceResult target, World world, BlockPos pos, EntityPlayer player) {
+      return com.anatawa12.fixRtm.rtm.block.BlockOrnamentMain.getPickBlock(world, pos, ItemInstalledObject.IstlObjType.RAILLOAD_SIGN);
    }
 }
