--- a/jp/ngt/rtm/block/BlockMechanism.java
+++ b/jp/ngt/rtm/block/BlockMechanism.java
@@ -29,6 +29,11 @@
          return true;
       } else {
          return false;
       }
    }
+
+   @Override
+   public net.minecraft.item.ItemStack getPickBlock(net.minecraft.block.state.IBlockState state, net.minecraft.util.math.RayTraceResult target, World world, net.minecraft.util.math.BlockPos pos, net.minecraft.entity.player.EntityPlayer player) {
+      return com.anatawa12.fixRtm.rtm.block.BlockOrnamentMain.getPickBlock(world, pos, jp.ngt.rtm.item.ItemInstalledObject.IstlObjType.MECHANISM);
+   }
 }
