--- a/jp/ngt/rtm/block/BlockPlant.java
+++ b/jp/ngt/rtm/block/BlockPlant.java
@@ -12,6 +12,16 @@
    }
 
    public TileEntity createNewTileEntity(World world, int par2) {
       return new TileEntityPlantOrnament();
    }
+
+   @Override
+   protected boolean onBlockActivated(jp.ngt.ngtlib.block.BlockArgHolder holder, float hitX, float hitY, float hitZ) {
+      return com.anatawa12.fixRtm.rtm.block.BlockOrnamentMain.onBlockActivated(holder);
+   }
+
+   @Override
+   public net.minecraft.item.ItemStack getPickBlock(net.minecraft.block.state.IBlockState state, net.minecraft.util.math.RayTraceResult target, World world, net.minecraft.util.math.BlockPos pos, net.minecraft.entity.player.EntityPlayer player) {
+      return com.anatawa12.fixRtm.rtm.block.BlockOrnamentMain.getPickBlock(world, pos, jp.ngt.rtm.item.ItemInstalledObject.IstlObjType.PLANT);
+   }
 }
