--- a/jp/ngt/rtm/block/BlockMachineBase.java
+++ b/jp/ngt/rtm/block/BlockMachineBase.java
@@ -22,10 +22,26 @@
       this.clickMachine(holder.getWorld(), holder.getBlockPos().getX(), holder.getBlockPos().getY(), holder.getBlockPos().getZ(), holder.getPlayer());
       return true;
    }
 
    protected boolean clickMachine(World world, int x, int y, int z, EntityPlayer player) {
+      if (world.isRemote) {
+         if (jp.ngt.ngtlib.util.NGTUtil.isEquippedItem(player, jp.ngt.rtm.RTMItem.installedObject)) {
+            net.minecraft.item.ItemStack itemStack = player.inventory.getCurrentItem();
+            TileEntityMachineBase machine = (TileEntityMachineBase) world.getTileEntity(new BlockPos(x, y, z));
+            assert machine != null;
+            if (com.anatawa12.fixRtm.UtilsKt.isItemOf(itemStack, machine)) {
+               com.anatawa12.fixRtm.UtilsKt.openGui(player, com.anatawa12.fixRtm.gui.GuiId.ChangeOffset, player.world, x, y, z);
+               return true;
+            }
+         }
+
+         if (player.isSneaking()) {
+            player.openGui(RTMCore.instance, RTMCore.guiIdSelectTileEntityModel, player.world, x, y, z);
+            return true;
+         }
+      }
       if (player.isSneaking()) {
          if (world.isRemote) {
             player.openGui(RTMCore.instance, RTMCore.guiIdSelectTileEntityModel, player.getEntityWorld(), x, y, z);
          }
 
@@ -33,12 +49,15 @@
       } else {
          return false;
       }
    }
 
+   @Deprecated
    public int getLightValue(IBlockState state, IBlockAccess world, BlockPos pos) {
-      TileEntityMachineBase tileentitymachinebase = (TileEntityMachineBase)world.getTileEntity(pos);
+      net.minecraft.tileentity.TileEntity tile = world.getTileEntity(pos);
+      if (!(tile instanceof TileEntityMachineBase)) return super.getLightValue(state);
+      TileEntityMachineBase tileentitymachinebase = (TileEntityMachineBase)tile;
       if (tileentitymachinebase == null) {
          return 0;
       } else {
          MachineConfig machineconfig = (MachineConfig)((ModelSetMachine)tileentitymachinebase.getResourceState().getResourceSet()).getConfig();
          return tileentitymachinebase.isGettingPower ? machineconfig.brightness[1] : machineconfig.brightness[0];
