--- a/jp/ngt/rtm/rail/util/RailMaker.java
+++ b/jp/ngt/rtm/rail/util/RailMaker.java
@@ -4,26 +4,30 @@
 import java.util.List;
 import jp.ngt.ngtlib.io.NGTLog;
 import net.minecraft.world.World;
 
 public final class RailMaker {
+   // see RailMapBasic.fixRTMRailMapVersion
+   public final int fixRTMRailMapVersion;
    private World worldObj;
    private List<RailPosition> rpList;
 
    public RailMaker(World world, List<RailPosition> par2) {
+      this(world, par2, RailMapBasic.fixRTMRailMapVersionCurrent);
+   }
+   public RailMaker(World world, List<RailPosition> par2, int fixRTMRailMapVersion) {
       this.worldObj = world;
       this.rpList = par2;
+      this.fixRTMRailMapVersion = fixRTMRailMapVersion;
    }
 
    public RailMaker(World world, RailPosition[] par2) {
-      this.worldObj = world;
-      this.rpList = new ArrayList<>();
-
-      for(RailPosition railposition : par2) {
-         this.rpList.add(railposition);
-      }
+      this(world, par2, RailMapBasic.fixRTMRailMapVersionCurrent);
+   }
 
+   public RailMaker(World world, RailPosition[] par2, int fixRTMRailMapVersion) {
+      this(world, new ArrayList<>(java.util.Arrays.asList(par2)), fixRTMRailMapVersion);
    }
 
    private SwitchType getSwitchType() {
       if (this.rpList.size() == 3) {
          int i = 0;
@@ -79,13 +83,8 @@
          if (switchtype.init(list, list1)) {
             return switchtype;
          }
       }
 
-      if (this.worldObj != null && !this.worldObj.isRemote) {
-         RailPosition railposition1 = this.rpList.get(0);
-         NGTLog.sendChatMessageToAll("message.rail.switch_type", new Object[]{railposition1.blockX, railposition1.blockY, railposition1.blockZ});
-      }
-
       return null;
    }
 }
