--- a/jp/ngt/rtm/rail/util/RailMapSwitch.java
+++ b/jp/ngt/rtm/rail/util/RailMapSwitch.java
@@ -3,12 +3,18 @@
 public final class RailMapSwitch extends RailMapBasic {
    public final RailDir startDir;
    public final RailDir endDir;
    private boolean isOpen;
 
+   /** @deprecated use {@link #RailMapSwitch(RailPosition, RailPosition, RailDir, RailDir, int)} */
+   @Deprecated
    public RailMapSwitch(RailPosition par1, RailPosition par2, RailDir sDir, RailDir eDir) {
-      super(par1, par2);
+      this(par1, par2, sDir, eDir, 0);
+      com.anatawa12.fixRtm.Deprecation.found("RailMapSwitch#RailMapSwitch");
+   }
+   public RailMapSwitch(RailPosition par1, RailPosition par2, RailDir sDir, RailDir eDir, int version) {
+      super(par1, par2, version);
       this.startDir = sDir;
       this.endDir = eDir;
    }
 
    /** @deprecated */
