--- a/jp/ngt/ngtlib/renderer/model/ModelLoader.java
+++ b/jp/ngt/ngtlib/renderer/model/ModelLoader.java
@@ -17,10 +17,14 @@
    public static PolygonModel loadModel(String path, VecAccuracy par1, Object... args) {
       return loadModel(new ResourceLocationCustom("minecraft", path), par1, args);
    }
 
    public static PolygonModel loadModel(ResourceLocation resource, VecAccuracy par1, Object... args) {
+      return com.anatawa12.fixRtm.ngtlib.renderer.model.ModelLoaderKt.loadModel(resource, par1, args);
+   }
+
+   public static PolygonModel loadModel__NGTLIB(ResourceLocation resource, VecAccuracy par1, Object... args) {
       String s = resource.toString();
 
       try {
          InputStream inputstream = NGTFileLoader.getInputStream(resource);
          if (FileType.OBJ.match(resource.getPath())) {
