--- a/jp/ngt/ngtlib/renderer/model/PolygonModel.java
+++ b/jp/ngt/ngtlib/renderer/model/PolygonModel.java
@@ -15,11 +15,11 @@
 
 @SideOnly(Side.CLIENT)
 public abstract class PolygonModel implements IModelNGT {
    protected String fileName;
    protected int drawMode;
-   protected VecAccuracy accuracy;
+   public VecAccuracy accuracy;
    protected float[] sizeBox = new float[6];
    public final List<Vertex> vertices = new ArrayList<Vertex>(1024);
    public final List<GroupObject> groupObjects = new ArrayList<GroupObject>(16);
    protected GroupObject currentGroupObject;
    int lineCount;
@@ -93,11 +93,11 @@
 
    public final float[] getSize() {
       return this.sizeBox;
    }
 
-   public void renderAll(boolean smoothing) {
+   public final void renderAll(boolean smoothing) {
       if (smoothing) {
          GL11.glShadeModel(7425);
       }
 
       NGTTessellator ngttessellator = NGTTessellator.instance;
@@ -108,18 +108,18 @@
          GL11.glShadeModel(7424);
       }
 
    }
 
-   public void tessellateAll(NGTTessellator tessellator, boolean smoothing) {
+   public final void tessellateAll(NGTTessellator tessellator, boolean smoothing) {
       for(GroupObject groupobject : this.groupObjects) {
          groupobject.render(tessellator, smoothing);
       }
 
    }
 
-   public void renderOnly(boolean smoothing, String... groupNames) {
+   public final void renderOnly(boolean smoothing, String... groupNames) {
       if (smoothing) {
          GL11.glShadeModel(7425);
       }
 
       for(GroupObject groupobject : this.groupObjects) {
@@ -134,11 +134,11 @@
          GL11.glShadeModel(7424);
       }
 
    }
 
-   public void renderPart(boolean smoothing, String partName) {
+   public final void renderPart(boolean smoothing, String partName) {
       if (smoothing) {
          GL11.glShadeModel(7425);
       }
 
       for(GroupObject groupobject : this.groupObjects) {
@@ -152,15 +152,15 @@
          GL11.glShadeModel(7424);
       }
 
    }
 
-   public int getDrawMode() {
+   public final int getDrawMode() {
       return this.drawMode;
    }
 
-   public List<GroupObject> getGroupObjects() {
+   public final List<GroupObject> getGroupObjects() {
       return this.groupObjects;
    }
 
    protected final float getFloat(String s) {
       try {
