--- a/jp/ngt/rtm/rail/BlockMarker.java
+++ b/jp/ngt/rtm/rail/BlockMarker.java
@@ -39,10 +39,11 @@
 import net.minecraft.world.World;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
 public class BlockMarker extends BlockContainerCustomWithMeta {
+   public static final org.apache.logging.log4j.Logger LOGGER = org.apache.logging.log4j.LogManager.getLogger("rtm.BlockMarker");
    public final BlockMarker.MarkerType markerType;
 
    public BlockMarker(BlockMarker.MarkerType type) {
       super(Material.GLASS);
       this.markerType = type;
@@ -84,10 +85,11 @@
    public static int getFacing(EntityLivingBase placer, boolean isDiagonal) {
       return isDiagonal ? NGTMath.floor(NGTMath.normalizeAngle((double)placer.rotationYaw + 180.0D) / 90.0D) & 3 : NGTMath.floor(NGTMath.normalizeAngle((double)placer.rotationYaw + 180.0D) / 90.0D + 0.5D) & 3;
    }
 
    public void onBlockPlacedBy(World world, BlockPos pos, IBlockState state, EntityLivingBase placer, ItemStack stack) {
+      BlockMarker.LOGGER.trace("Placed block ({} at {}", markerType, pos);
       int i = stack.getItemDamage();
       int j = getFacing(placer, i >= 4);
       int k = i / 4;
       BlockUtil.setBlock(world, pos, this, j + k * 4, 2);
    }
@@ -125,10 +127,11 @@
    public void makeRailMap(TileEntityMarker marker, int x, int y, int z) {
       this.onMarkerActivated(marker.getWorld(), x, y, z, (EntityPlayer)null, false);
    }
 
    public boolean onMarkerActivated(World world, int x, int y, int z, EntityPlayer player, boolean makeRail) {
+      com.anatawa12.fixRtm.rtm.HooksKt.BlockMarker_onMarkerActivated(player);
       ResourceStateRail resourcestaterail = this.hasRail(player, makeRail);
       if (resourcestaterail == null) {
          return false;
       } else {
          boolean flag = player == null || player.capabilities.isCreativeMode;
@@ -141,41 +144,33 @@
          return createRail(world, x, y, z, list, resourcestaterail, makeRail, flag);
       }
    }
 
    private List<RailPosition> searchAllMarker(World world, int x, int y, int z) {
-      List<RailPosition> list = new ArrayList<>();
       int i = RTMCore.railGeneratingDistance;
       int j = i * 2;
+      int dis3 = j * j;
       int k = RTMCore.railGeneratingHeight;
       int l = k * 2;
 
-      for(int i1 = 0; i1 < j; ++i1) {
-         for(int j1 = 0; j1 < l; ++j1) {
-            for(int k1 = 0; k1 < j; ++k1) {
-               int l1 = x - i + i1;
-               int i2 = y - k + j1;
-               int j2 = z - i + k1;
-               RailPosition railposition = this.getRailPosition(world, l1, i2, j2);
-               if (railposition != null) {
-                  list.add(railposition);
-               }
-            }
-         }
-      }
+      List<RailPosition> list = world.loadedTileEntityList.stream()
+         .filter(TileEntityMarker.class::isInstance)
+         .map(TileEntityMarker.class::cast)
+         .filter(tile -> tile.getPos().distanceSqToCenter(x, tile.getY(), z) < dis3)
+         .filter(tile -> Math.abs(tile.getY() - y) < k)
+         .map(TileEntityMarker::getMarkerRP)
+         .filter(java.util.Objects::nonNull)
+         .sorted(java.util.Comparator.<RailPosition>comparingInt(v -> v.switchType)
+               .thenComparingInt(v -> v.blockY)
+               .thenComparingInt(Object::hashCode))
+         .collect(java.util.stream.Collectors.toList());
 
-      list.sort((arg0, arg1) -> {
-         if (arg0.switchType != arg1.switchType) {
-            return arg1.switchType - arg0.switchType;
-         } else {
-            return arg0.blockY != arg1.blockY ? arg0.blockY - arg1.blockY : arg0.hashCode() - arg1.hashCode();
-         }
-      });
       return list;
    }
 
    public static boolean createRail(World world, int x, int y, int z, List<RailPosition> rps, ResourceStateRail state, boolean makeRail, boolean isCreative) {
+      LOGGER.trace("making rail via {}", rps);
       if (rps.size() == 1) {
          RailPosition railposition = rps.get(0);
          if (railposition.hasScript()) {
             createCustomRail(world, railposition, state, makeRail, isCreative);
          }
@@ -190,10 +185,11 @@
             createNormalRail(world, railposition2, railposition3, state, makeRail, isCreative);
          }
       } else if (rps.size() > 2) {
          createSwitchRail(world, x, y, z, rps, state, makeRail, isCreative);
       }
+      LOGGER.trace("made rail via {}", rps);
 
       return false;
    }
 
    private static boolean createNormalRail(World world, RailPosition start, RailPosition end, ResourceStateRail prop, boolean makeRail, boolean isCreative) {
@@ -292,18 +288,18 @@
                tileentitylargerailswitchcore.sendPacket();
                return true;
             } else {
                TileEntity tileentity = BlockUtil.getTileEntity(world, x, y, z);
                if (tileentity instanceof TileEntityMarker) {
-                  List<BlockPos> list = new ArrayList();
+                  List<BlockPos> list1 = new ArrayList();
 
                   for(int i = 0; i < list.size(); ++i) {
                      RailPosition railposition2 = list.get(i);
-                     list.add(new BlockPos(railposition2.blockX, railposition2.blockY, railposition2.blockZ));
+                     list1.add(new BlockPos(railposition2.blockX, railposition2.blockY, railposition2.blockZ));
                   }
 
-                  ((TileEntityMarker)tileentity).setMarkersPos(list);
+                  ((TileEntityMarker)tileentity).setMarkersPos(list1);
                }
 
                return false;
             }
          }
