--- a/jp/ngt/rtm/rail/BlockMarker.java
+++ b/jp/ngt/rtm/rail/BlockMarker.java
@@ -124,10 +124,11 @@
    public void makeRailMap(TileEntityMarker marker, int x, int y, int z) {
       this.onMarkerActivated(marker.getWorld(), x, y, z, (EntityPlayer)null, false);
    }
 
    public boolean onMarkerActivated(World world, int x, int y, int z, EntityPlayer player, boolean makeRail) {
+      com.anatawa12.fixRtm.rtm.HooksKt.BlockMarker_onMarkerActivated(player);
       ResourceStateRail resourcestaterail = this.hasRail(player, makeRail);
       if (resourcestaterail == null) {
          return false;
       } else {
          boolean flag = player == null || player.capabilities.isCreativeMode;
