--- a/jp/ngt/rtm/rail/BlockLargeRailBase.java
+++ b/jp/ngt/rtm/rail/BlockLargeRailBase.java
@@ -179,11 +179,11 @@
 
          Block block = tileentitylargerailcore.getResourceState().block;
          boolean flag = block.getMaterial(block.getDefaultState()) == Material.SNOW;
          if (flag && entity instanceof EntityBogie) {
             EntityTrainBase entitytrainbase = ((EntityBogie)entity).getTrain();
-            if (entitytrainbase != null && entitytrainbase.getSpeed() > 0.0F) {
+            if (entitytrainbase != null && Math.abs(entitytrainbase.getSpeed()) > 0.0F) {
                double d0 = (double)entitytrainbase.getSpeed() * 0.125D;
 
                for(int i = 0; i < 5; ++i) {
                   double d1 = (double)pos.getX() + (double)world.rand.nextFloat();
                   double d2 = (double)pos.getY() + (double)world.rand.nextFloat() * 0.25D;
@@ -212,11 +212,11 @@
       return tileentity instanceof TileEntityLargeRailBase ? ((TileEntityLargeRailBase)tileentity).getRailCore() : null;
    }
 
    public ItemStack getPickBlock(IBlockState state, RayTraceResult target, World world, BlockPos pos, EntityPlayer player) {
       TileEntityLargeRailCore tileentitylargerailcore = getCore(world, pos);
-      return tileentitylargerailcore != null ? ItemRail.copyItemFromRail(tileentitylargerailcore) : null;
+      return tileentitylargerailcore != null ? ItemRail.copyItemFromRail(tileentitylargerailcore) : ItemStack.EMPTY;
    }
 
    public PathNodeType getAiPathNodeType(IBlockState state, IBlockAccess world, BlockPos pos) {
       return this.preventMobMovement(world, pos) ? PathNodeType.BLOCKED : PathNodeType.RAIL;
    }
