--- a/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.java
+++ b/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.java
@@ -268,11 +268,10 @@
 
       try {
          ModelSetBase modelsetbase = this.getResourceState().getResourceSet();
          axisalignedbb = modelsetbase.getCollisionObj().getSizeBox();
       } catch (NullPointerException nullpointerexception) {
-         nullpointerexception.printStackTrace();
       }
 
       if (axisalignedbb == null) {
          axisalignedbb = new AxisAlignedBB(-1.5D, 0.0D, -2.0D, 1.5D, 3.0D, 2.0D);
       }
@@ -532,21 +531,33 @@
    public void setTrainStateData(int id, byte data) {
       this.setVehicleState(TrainState.getStateType(id), data);
    }
 
    public byte getVehicleState(TrainState.TrainStateType type) {
-      return (byte)this.getResourceState().getDataMap().getInt(type.toString());
+      return type.clap((byte)this.getResourceState().getDataMap().getInt(type.toString()), this);
    }
 
    public void setVehicleState(TrainState.TrainStateType type, byte data) {
-      this.getResourceState().getDataMap().setInt(type.toString(), data, 3);
+      this.getResourceState().getDataMap().setInt(type.toString(), type.clap(data, this), 3);
    }
 
    @SideOnly(Side.CLIENT)
    public void syncVehicleState(TrainState.TrainStateType type, byte data) {
       RTMCore.NETWORK_WRAPPER.sendToServer(new PacketSetTrainState(this, type, data));
    }
 
    public VehicleNGTO getNGTO() {
       return null;
    }
+
+   @Override
+   public void onRemovedFromWorld() {
+      super.onRemovedFromWorld();
+      com.anatawa12.fixRtm.rtm.entity.vehicle.EntityVehicleBaseKt.onRemovedFromWorld(this);
+   }
+
+   @Override
+    public void addEntityCrashInfo(net.minecraft.crash.CrashReportCategory category) {
+        super.addEntityCrashInfo(category);
+        com.anatawa12.fixRtm.rtm.entity.vehicle.EntityVehicleBaseKt.addEntityCrashInfo(this, category);
+    }
 }
