--- a/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.java
+++ b/jp/ngt/rtm/entity/vehicle/EntityVehicleBase.java
@@ -268,13 +268,13 @@
 
       try {
          ModelSetBase modelsetbase = (ModelSetBase)this.getResourceState().getResourceSet();
          axisalignedbb = modelsetbase.getCollisionObj().getSizeBox();
       } catch (NullPointerException nullpointerexception) {
-         nullpointerexception.printStackTrace();
-         axisalignedbb = new AxisAlignedBB(-1.5D, 0.0D, -2.0D, 1.5D, 3.0D, 2.0D);
       }
+      if (axisalignedbb == null)
+         axisalignedbb = new AxisAlignedBB(-1.5D, 0.0D, -2.0D, 1.5D, 3.0D, 2.0D);
 
       Vec3 vec3 = PooledVec3.create(ridingEntity.posX - this.posX, ridingEntity.posY - this.posY, ridingEntity.posZ - this.posZ);
       vec3 = vec3.rotateAroundY(-this.rotationYaw);
       float f = NGTMath.wrapAngle(-passenger.rotationYaw - this.rotationYaw);
       double d0 = (double)f >= 0.0D ? axisalignedbb.maxX + 0.5D : axisalignedbb.minX - 0.5D;
@@ -544,6 +544,18 @@
    }
 
    public VehicleNGTO getNGTO() {
       return null;
    }
+
+   @Override
+   public void onRemovedFromWorld() {
+      super.onRemovedFromWorld();
+      com.anatawa12.fixRtm.rtm.entity.vehicle.EntityVehicleBaseKt.onRemovedFromWorld(this);
+   }
+
+   @Override
+    public void addEntityCrashInfo(net.minecraft.crash.CrashReportCategory category) {
+        super.addEntityCrashInfo(category);
+        com.anatawa12.fixRtm.rtm.entity.vehicle.EntityVehicleBaseKt.addEntityCrashInfo(this, category);
+    }
 }
