--- a/jp/ngt/rtm/entity/vehicle/RenderVehicleBase.java
+++ b/jp/ngt/rtm/entity/vehicle/RenderVehicleBase.java
@@ -51,11 +51,11 @@
       GL11.glRotatef(f, 0.0F, 1.0F, 0.0F);
       float f1 = vehicle.prevRotationPitch + (vehicle.rotationPitch - vehicle.prevRotationPitch) * par9;
       GL11.glRotatef(-f1, 1.0F, 0.0F, 0.0F);
       float f2 = vehicle.prevRotationRoll + (vehicle.rotationRoll - vehicle.prevRotationRoll) * par9;
       GL11.glRotatef(f2, 0.0F, 0.0F, 1.0F);
-      if (Minecraft.getMinecraft().gameSettings.showDebugInfo) {
+      if (Minecraft.getMinecraft().getRenderManager().isDebugBoundingBox()) {
          this.debugCollision(vehicle);
       }
 
       VehicleNGTO vehiclengto = vehicle.getNGTO();
       if (vehiclengto != null) {
